<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="HE Yunye">
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>PCA - GWASTutorial</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">GWASTutorial</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="../02_Linux_basics/" class="nav-link">Linux</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Pre-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../01_Dataset/" class="dropdown-item">Sample Dataset</a>
</li>
                                    
<li>
    <a href="../03_Data_formats/" class="dropdown-item">Data Formats</a>
</li>
                                    
<li>
    <a href="../04_Data_QC/" class="dropdown-item">Data QC</a>
</li>
                                    
<li>
    <a href="./" class="dropdown-item active">PCA</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../06_Association_tests/" class="dropdown-item">Association tests</a>
</li>
                                    
<li>
    <a href="../Visualization/" class="dropdown-item">Visualization</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Post-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../07_Annotation/" class="dropdown-item">Annotation</a>
</li>
                                    
<li>
    <a href="../08_LDSC/" class="dropdown-item">LDSC</a>
</li>
                                    
<li>
    <a href="../09_Gene_based_analysis/" class="dropdown-item">Gene/Gene-set tests</a>
</li>
                                    
<li>
    <a href="../10_PRS/" class="dropdown-item">PRS</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Others <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../90_Recommended_Reading/" class="dropdown-item">Recommended reading</a>
</li>
                                    
<li>
    <a href="../80_anaconda/" class="dropdown-item">Anaconda</a>
</li>
                                    
<li>
    <a href="../81_jupyter_notebook/" class="dropdown-item">Jupyter notebook</a>
</li>
                                    
<li>
    <a href="../82_windows_linux_subsystem/" class="dropdown-item">WLS</a>
</li>
                                    
<li>
    <a href="../83_git_and_github/" class="dropdown-item">Git and github</a>
</li>
                                    
<li>
    <a href="../60_awk/" class="dropdown-item">awk</a>
</li>
                                    
<li>
    <a href="../61_sed/" class="dropdown-item">sed</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                            <li class="nav-item">
                                <a rel="prev" href="../04_Data_QC/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../06_Association_tests/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/Cloufield/GWASTutorial" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#principle-conponent-analysis-pca" class="nav-link">Principle conponent analysis (PCA)</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#preparation" class="nav-link">Preparation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#pca-steps" class="nav-link">PCA steps</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#sample-codes" class="nav-link">Sample codes</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#plotting-the-pcs" class="nav-link">Plotting the PCs</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#pca-umap" class="nav-link">PCA-UMAP</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
            
            <li class="nav-item" data-level="1"><a href="#references" class="nav-link">References</a>
              <ul class="nav flex-column">
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="principle-conponent-analysis-pca">Principle conponent analysis (PCA)</h1>
<p>PCA is by far the most commonly used dimension reduction approach used in population genetics which could identify the diffenrence in ancestry among the sample individuals.
For GWAS we also need to include top PCs to adjust for the population stratification.</p>
<p>Please read the following paper on how we apply PCA to genetic data:
Price, A., Patterson, N., Plenge, R. et al. Principal components analysis corrects for stratification in genome-wide association studies. Nat Genet 38, 904–909 (2006). https://doi.org/10.1038/ng1847 https://www.nature.com/articles/ng1847</p>
<p>Simply speaking, GRM (genetic relationship matrix) is first estimated and then PCA is applied to this matrix to generate PCs for each individual.</p>
<p>So before association analysis, we will learn how to run PCA analysis first.</p>
<ul>
<li><a href="#preparation">Preparation</a></li>
<li><a href="#pca-steps">PCA steps</a></li>
<li><a href="#sample-codes">Sample codes</a></li>
<li><a href="#plotting-the-pcs">Plotting the PCs</a></li>
<li><a href="#pca-umap">PCA-UMAP</a></li>
<li><a href="#references">References</a></li>
</ul>
<h2 id="preparation">Preparation</h2>
<h3 id="exclude-snps-in-high-ld-or-hla-regions">Exclude SNPs in high-LD or HLA regions</h3>
<p>For PCA, we first exclude SNPs in high-LD or HLA regions from the genotype data. </p>
<p>Note: the reason why we want to exclude such high-LD or HLA regions is described in:
- Price, A. L., Weale, M. E., Patterson, N., Myers, S. R., Need, A. C., Shianna, K. V., Ge, D., Rotter, J. I., Torres, E., Taylor, K. D., Goldstein, D. B., &amp; Reich, D. (2008). Long-range LD can confound genome scans in admixed populations. American journal of human genetics, 83(1), 132–139. https://doi.org/10.1016/j.ajhg.2008.06.005 </p>
<h3 id="download-bed-like-files-for-high-ld-or-hla-regions">Download BED-like files for high-LD or HLA regions</h3>
<p>You can simply copy the list of high-LD or HLA regions in Genome build version(.bed format) to a text file <code>high-ld.txt</code>. For details, please check https://genome.sph.umich.edu/wiki/Regions_of_high_linkage_disequilibrium_(LD).</p>
<pre><code>$cat high-ld-hg19.txt 
1   48000000    52000000    highld
2   86000000    100500000   highld
2   134500000   138000000   highld
2   183000000   190000000   highld
3   47500000    50000000    highld
3   83500000    87000000    highld
3   89000000    97500000    highld
5   44500000    50500000    highld
5   98000000    100500000   highld
5   129000000   132000000   highld
5   135500000   138500000   highld
6   25000000    35000000    highld
6   57000000    64000000    highld
6   140000000   142500000   highld
7   55000000    66000000    highld
8   7000000 13000000    highld
8   43000000    50000000    highld
8   112000000   115000000   highld
10  37000000    43000000    highld
11  46000000    57000000    highld
11  87500000    90500000    highld
12  33000000    40000000    highld
12  109500000   112000000   highld
20  32000000    34500000    highld
</code></pre>
<h3 id="create-a-list-for-snps-in-high-ld-or-hla-regions">Create a list for SNPs in high-LD or HLA regions</h3>
<p>Next, use <code>high-ld.txt</code> to extract all SNPs which are located in the regions described in the file using the code as follows:</p>
<pre><code>plink --file ${plinkFile} --make-set high-ld.txt --write-set --out hild
</code></pre>
<p>For example:</p>
<pre><code>plinkFile=&quot;../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot; #!please set this to your own path

plink \
    --bfile ${plinkFile} \
    --make-set high-ld-hg19.txt \
    --write-set \
    --out hild
</code></pre>
<p>And all SNPs in the regions will be extracted to hild.set.</p>
<pre><code>$head hild.set
highld
1:48000156:C:G
1:48002096:C:G
1:48003081:T:C
1:48004776:C:T
1:48006500:A:G
1:48006546:C:T
1:48008102:T:G
1:48009994:C:T
1:48009997:C:A
</code></pre>
<p>For downstream analysis, we can exclude these SNPs using <code>--exclude hild.set</code>.</p>
<hr />
<h2 id="pca-steps">PCA steps</h2>
<ul>
<li>1.Pruning (https://www.cog-genomics.org/plink/2.0/ld#indep)</li>
<li>2.Removing relatives (usually 2-degree) (https://www.cog-genomics.org/plink/2.0/distance#king_cutoff)</li>
<li>3.Run PCA using un-related samples and independent SNPs (https://www.cog-genomics.org/plink/2.0/strat#pca)</li>
<li>4.Project to all samples (https://www.cog-genomics.org/plink/2.0/score#pca_project)</li>
</ul>
<hr />
<h2 id="sample-codes">Sample codes</h2>
<pre><code>plinkFile=&quot;&quot; #please set this to your own path
outPrefix=&quot;plink_results&quot;
threadnum=2
hildset = hild.set 

# LD-pruning, excluding high-LD and HLA regions
plink2 \
        --bfile ${plinkFile} \
    --threads ${threadnum} \
    --exclude ${hildset} \ 
    --indep-pairwise 500 50 0.2 \
        --out ${outPrefix}

# Remove related samples using king-cuttoff
plink2 \
        --bfile ${plinkFile} \
    --extract ${outPrefix}.prune.in \
        --king-cutoff 0.0884 \
    --threads ${threadnum} \
        --out ${outPrefix}

# PCA after pruning and removing related samples
plink2 \
        --bfile ${plinkFile} \
        --keep ${outPrefix}.king.cutoff.in.id \
    --extract ${outPrefix}.prune.in \
    --freq counts \
    --threads ${threadnum} \
        --pca approx allele-wts \
        --out ${outPrefix}

# Projection (related and unrelated samples)
plink2 \
        --bfile ${plinkFile} \
    --threads ${threadnum} \
        --read-freq ${outPrefix}.acount \
    --score ${outPrefix}.eigenvec.allele 2 5 header-read no-mean-imputation variance-standardize \
        --score-col-nums 6-15 \
        --out ${outPrefix}_projected
</code></pre>
<p>After step 3, the 'allele-wts' modifier requests an additional one-line-per-allele .eigenvec.allele file with PCs expressed as allele weights instead of sample weights.</p>
<p>We will get the <code>plink_results.eigenvec.allele</code> file, which will be used to project onto all samples along with a allele count <code>plink_results.acount</code> file.</p>
<pre><code>$head plink_results.eigenvec.allele
#CHROM  ID  REF ALT A1  PC1 PC2 PC3 PC4 PC5 PC6 PC7 PC8 PC9 PC10
1   1:13273:G:C G   C   G   1.12369 -0.320826   -0.0206569  -0.218665   0.869801    0.378433    -0.0723841  -0.227555   0.0361673   -0.368192
1   1:13273:G:C G   C   C   -1.12369    0.320826    0.0206569   0.218665    -0.869801   -0.378433   0.0723841   0.227555    -0.0361673  0.368192
1   1:14599:T:A T   A   T   0.99902 -1.15824    -1.80519    -0.36774    0.179881    0.25242 0.068899    0.206564    -0.342483   0.103762
1   1:14599:T:A T   A   A   -0.99902    1.15824 1.80519 0.36774 -0.179881   -0.25242    -0.068899   -0.206564   0.342483    -0.103762
1   1:14930:A:G A   G   A   -0.0704343  -0.35091    -0.41535    -0.304856   0.081039    -0.49408    -0.0667606  -0.0698847  0.245836    0.330869
1   1:14930:A:G A   G   G   0.0704343   0.35091 0.41535 0.304856    -0.081039   0.49408 0.0667606   0.0698847   -0.245836   -0.330869
1   1:69897:T:C T   C   T   -0.514024   0.563153    -0.997768   -0.298234   -0.840608   -0.247155   0.545471    -0.675274   -0.787836   -0.509647
1   1:69897:T:C T   C   C   0.514024    -0.563153   0.997768    0.298234    0.840608    0.247155    -0.545471   0.675274    0.787836    0.509647
1   1:86331:A:G A   G   A   -0.169641   -0.0125126  -0.531174   -0.0219291  0.614439    0.140143    0.133833    -0.570109   0.392805    -0.065334

#head plink_results.acount
#CHROM  ID  REF ALT ALT_CTS OBS_CT
1   1:13273:G:C G   C   63  1004
1   1:14599:T:A T   A   90  1004
1   1:14930:A:G A   G   417 1004
1   1:69897:T:C T   C   879 1004
1   1:86331:A:G A   G   87  1004
1   1:91581:G:A G   A   499 1004
1   1:122872:T:G    T   G   259 1004
1   1:135163:C:T    C   T   91  1004
1   1:233473:C:G    C   G   156 1004

</code></pre>
<p>Please check https://www.cog-genomics.org/plink/2.0/score#pca_project for more details on projection.
Eventually, we will get the PCA results for all samples.</p>
<pre><code>head plink_results_projected.sscore
#FID    IID ALLELE_CT   NAMED_ALLELE_DOSAGE_SUM PC1_AVG PC2_AVG PC3_AVG PC4_AVG PC5_AVG PC6_AVG PC7_AVG PC8_AVG PC9_AVG PC10_AVG
0   HG00403 219504  219504  0.000643981 -0.0297502  -0.0151499  -0.0122381  0.0229149   0.0235408   -0.033705   -0.0075127  -0.0125402  0.00271677
0   HG00404 219504  219504  -0.000492225    -0.031018   -0.00764244 -0.0204998  0.0284068   -0.00872449 0.0123353   -0.00492058 -0.00557003 0.0248966
0   HG00406 219504  219504  0.00620984  -0.034375   -0.00898555 -0.00335076 -0.0217559  -0.0182433  0.00333925  -0.00760613 -0.0340018  0.00641082
0   HG00407 219504  219504  0.00678586  -0.0239308  -0.00704419 -0.00466139 0.00985433  0.000889767 0.00679557  -0.0200495  -0.0131869  0.0350328
0   HG00409 219504  219504  -0.00236345 -0.0231604  0.0320665   0.0145563   0.0236768   0.00704788  0.012859    0.0319605   -0.0130627  0.0110219
0   HG00410 219504  219504  0.000670927 -0.0210665  0.0467767   0.00293079  0.0184061   0.045967    0.00384994  0.0212317   -0.0296434  0.0237174
0   HG00419 219504  219504  0.00526139  -0.0369818  -0.00974662 0.00855412  -0.0053907  -0.00102057 0.0063254   0.0140126   -0.00600854 0.00732882
0   HG00421 219504  219504  0.00038356  -0.0319534  -0.00648054 0.00311739  -0.022044   0.0064945   -0.0105273  -0.0276718  -0.00973368 0.0208449
0   HG00422 219504  219504  0.00437335  -0.0323416  -0.0111979  0.0106245   -0.0267334  0.00142919  -0.00487295 -0.0124099  -0.00467014 -0.0188086
</code></pre>
<h2 id="plotting-the-pcs">Plotting the PCs</h2>
<p>You can now create scatterplots of the PCs using R or python.</p>
<p>For plotting using python:
<a href="https://github.com/Cloufield/GWASTutorial/blob/main/05_PCA/plot_PCA.ipynb">plot_PCA.ipynb</a></p>
<p><img width="473" alt="image" src="https://user-images.githubusercontent.com/40289485/209298567-d4871fd0-aaa4-4d90-a7db-bc6aa34ab011.png">
Note : We only used 20% of all availble variants. This figure only very roughly shows the population structure in East Asia.</p>
<p>Requrements:
- python&gt;3
- numpy,pandas,seaborn,matplotlib</p>
<h2 id="pca-umap">PCA-UMAP</h2>
<p>(optional) 
We can also apply another non-linear dimension reduction algorithm called UMAP to the PCs to further identfy the local structures. (PCA-UMAP)</p>
<p>For more details, please check:
- https://umap-learn.readthedocs.io/en/latest/index.html</p>
<p>An example of PCA and PCA-UMAP for population genetics:
- Sakaue, S., Hirata, J., Kanai, M., Suzuki, K., Akiyama, M., Lai Too, C., ... &amp; Okada, Y. (2020). Dimensionality reduction reveals fine-scale structure in the Japanese population with consequences for polygenic risk prediction. Nature communications, 11(1), 1-11.</p>
<h1 id="references">References</h1>
<ul>
<li>(PCA) Price, A., Patterson, N., Plenge, R. et al. Principal components analysis corrects for stratification in genome-wide association studies. Nat Genet 38, 904–909 (2006). https://doi.org/10.1038/ng1847 https://www.nature.com/articles/ng1847</li>
<li>(why removing high-LD regions) Price, A. L., Weale, M. E., Patterson, N., Myers, S. R., Need, A. C., Shianna, K. V., Ge, D., Rotter, J. I., Torres, E., Taylor, K. D., Goldstein, D. B., &amp; Reich, D. (2008). Long-range LD can confound genome scans in admixed populations. American journal of human genetics, 83(1), 132–139. https://doi.org/10.1016/j.ajhg.2008.06.005 </li>
<li>(UMAP) McInnes, L., Healy, J., &amp; Melville, J. (2018). Umap: Uniform manifold approximation and projection for dimension reduction. arXiv preprint arXiv:1802.03426.</li>
<li>(UMAP in population genetics) Diaz-Papkovich, A., Anderson-Trocmé, L. &amp; Gravel, S. A review of UMAP in population genetics. J Hum Genet 66, 85–91 (2021). https://doi.org/10.1038/s10038-020-00851-4 https://www.nature.com/articles/s10038-020-00851-4</li>
<li>(king-cutoff) Manichaikul, A., Mychaleckyj, J. C., Rich, S. S., Daly, K., Sale, M., &amp; Chen, W. M. (2010). Robust relationship inference in genome-wide association studies. Bioinformatics, 26(22), 2867-2873.</li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>GWASTutorial is licensed under the MIT license</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>

        <div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
