<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="HE Yunye">
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Association tests - GWASTutorial</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">GWASTutorial</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="../02_Linux_basics/" class="nav-link">Linux</a>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Pre-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../01_Dataset/" class="dropdown-item">Sample Dataset</a>
</li>
                                    
<li>
    <a href="../03_Data_formats/" class="dropdown-item">Data Formats</a>
</li>
                                    
<li>
    <a href="../04_Data_QC/" class="dropdown-item">Data QC</a>
</li>
                                    
<li>
    <a href="../05_PCA/" class="dropdown-item">PCA</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="./" class="dropdown-item active">Association tests</a>
</li>
                                    
<li>
    <a href="../Visualization/" class="dropdown-item">Visualization</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Post-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../07_Annotation/" class="dropdown-item">Annotation</a>
</li>
                                    
<li>
    <a href="../08_LDSC/" class="dropdown-item">LDSC</a>
</li>
                                    
<li>
    <a href="../09_Gene_based_analysis/" class="dropdown-item">Gene/Gene-set tests</a>
</li>
                                    
<li>
    <a href="../10_PRS/" class="dropdown-item">PRS</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Others <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../80_anaconda/" class="dropdown-item">Anaconda</a>
</li>
                                    
<li>
    <a href="../81_jupyter_notebook/" class="dropdown-item">Jupyter notebook</a>
</li>
                                    
<li>
    <a href="../82_windows_linux_subsystem/" class="dropdown-item">WLS</a>
</li>
                                    
<li>
    <a href="../83_git_and_github/" class="dropdown-item">Git and github</a>
</li>
                                    
<li>
    <a href="../60_awk/" class="dropdown-item">awk</a>
</li>
                                    
<li>
    <a href="../61_sed/" class="dropdown-item">sed</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                            <li class="nav-item">
                                <a rel="prev" href="../05_PCA/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../Visualization/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/Cloufield/GWASTutorial" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#association-test" class="nav-link">Association test</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#file-preparation" class="nav-link">File preparation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#genetic-models" class="nav-link">Genetic models</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#association-testing" class="nav-link">Association testing</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#genomic-control" class="nav-link">Genomic control</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#significant-loci" class="nav-link">Significant loci</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
            
            <li class="nav-item" data-level="1"><a href="#visualization" class="nav-link">Visualization</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#mahanttan-plot-and-quantile-quantile-plot" class="nav-link">Mahanttan plot and Quantile-Quantile plot</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#regional-plot" class="nav-link">Regional plot</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="association-test">Association test</h1>
<h2 id="file-preparation">File preparation</h2>
<p>To perform genome-wide association study (GWAS), usually we need the following files:</p>
<ul>
<li><strong>Genotype file</strong> (or dosage file) : usually in PLINK format, VCF format, or BGEN format.</li>
<li><strong>Phenotype file</strong> : plain text file.</li>
<li><strong>Covariate file</strong> (optional): plain text file. Usually covaraites include age, sex, and top Pricipal Components. </li>
</ul>
<p>For example,</p>
<pre><code># Phenotype file for a simulated binary trait; B1 is the phenotype name; 1 means control, 2 means case.
head 1kgeas_binary.txt 
FID IID B1
0 HG00403 2 
0 HG00404 1 
0 HG00406 2 
0 HG00407 2 
0 HG00409 2 
0 HG00410 2 
0 HG00419 1 
0 HG00421 2 
0 HG00422 1

# Covariate file (only top PCs calculated in the previous PCA section)
#FID    IID ALLELE_CT   NAMED_ALLELE_DOSAGE_SUM PC1_AVG PC2_AVG PC3_AVG PC4_AVG PC5_AVG PC6_AVG PC7_AVG PC8_AVG PC9_AVG PC10_AVG
0   HG00403 224016  224016  0.000246109 0.0292717   -0.0127437  -0.0135105  0.0301317   0.0196699   0.0232392   -0.0205941  -0.00416543 0.0121819
0   HG00404 224016  224016  -0.000716664    0.032043    -0.00573731 -0.0189504  0.0277385   -0.0154478  -0.0136551  -0.00147269 0.00510851  0.0268694
0   HG00406 224016  224016  0.00681476  0.0346759   -0.00706221 -0.0054266  -0.025458   -0.0166505  -0.00510707 -0.00105581 -0.0224028  0.0202523
0   HG00407 224016  224016  0.00695106  0.0244759   -0.00890072 -0.000694057    0.00946838  -0.00773378 -0.0139923  -0.0204871  -0.0003338  0.0387022
0   HG00409 224016  224016  -0.0023481  0.0213108   0.03235 0.0158793   0.026655    0.00324455-0.0107152    0.0317714   -0.00764455 0.0155973
0   HG00410 224016  224016  0.000926147 0.0198139   0.0475798   0.00314974  0.0275373   0.041886    -0.0133896  0.0184717   -0.0143644  0.0291036
0   HG00419 224016  224016  0.00580767  0.0369208   -0.00907507 0.00903163  -0.00649345 -0.000472359    -0.00327011 0.0160456   -0.005133   0.0141021
0   HG00421 224016  224016  0.000987901 0.0336895   -0.00697814 0.00557199  -0.0210537  0.00700968  0.00319921  -0.0215999  0.00127686  0.0350116
0   HG00422 224016  224016  0.00440288  0.0335901   -0.0125043  0.0135621   -0.0228428  0.00492741  0.00445856  -0.00911147 -0.00312742 -0.00784459
</code></pre>
<h2 id="genetic-models">Genetic models</h2>
<p>To test the association between a phenotype and genotypes, we also need to group the genotypes based on genetic models.</p>
<p>There are three basic genetic models:
- Additive model (ADD)
- Dominant model (DOM)
- Recessive model (REC)</p>
<p>For example, suppose we have a biallelic SNP whose reference allele is A and alternative allele is G.</p>
<p>There are three possible genotypes for this SNP: AA, AG, and GG.</p>
<p>The table shows how we group different genotypes under each genetic model.</p>
<table>
<thead>
<tr>
<th>Genetic models</th>
<th>AA</th>
<th>AG</th>
<th>GG</th>
</tr>
</thead>
<tbody>
<tr>
<td>Additive model</td>
<td>0</td>
<td>1</td>
<td>2</td>
</tr>
<tr>
<td>Dominant model</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr>
<td>Recessive model</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>
</tbody>
</table>
<h2 id="association-testing">Association testing</h2>
<p>Please check https://www.cog-genomics.org/plink/2.0/assoc for more details.</p>
<p>We will perform a logistic regression with firth correction for a simulated binary trait under additive model using the 1KG East Asian individuals.</p>
<p>Sample code:</p>
<pre><code>genotypeFile=&quot;../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot;  #!please set this to your own path
phenotypeFile=&quot;../01_Dataset/1kgeas_binary.txt&quot; #!please set this to your own path
covariateFile=&quot;../05_PCA/plink_results_projected.sscore&quot;

covariateCols=6-10
colName=&quot;B1&quot;
threadnum=2

plink2 \
    --bfile ${genotypeFile} \
    --pheno ${phenotypeFile} \
    --pheno-name ${colName} \
    --maf 0.01 \
    --covar ${covariateFile} \
    --covar-col-nums ${covariateCols} \
    --glm hide-covar firth \
    --threads ${threadnum} \
    --out 1kgeas
</code></pre>
<p>You will see a similar log like:</p>
<pre><code>PLINK v2.00a3LM 64-bit Intel (12 Dec 2020)     www.cog-genomics.org/plink/2.0/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to 1kgeas.log.
Options in effect:
  --bfile ../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020
  --covar ../05_PCA/plink_results_projected.sscore
  --covar-col-nums 6-10
  --glm hide-covar firth
  --maf 0.01
  --out 1kgeas
  --pheno ../01_Dataset/1kgeas_binary.txt
  --pheno-name B1
  --threads 2

Start time: Tue Dec 27 23:02:52 2022
15957 MiB RAM detected; reserving 7978 MiB for main workspace.
Using up to 2 compute threads.
504 samples (0 females, 0 males, 504 ambiguous; 504 founders) loaded from
../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam.
1122299 variants loaded from
../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim.
1 binary phenotype loaded (201 cases, 302 controls).
5 covariates loaded from ../05_PCA/plink_results_projected.sscore.
Calculating allele frequencies... done.
0 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
1122299 variants remaining after main filters.
--glm Firth regression on phenotype 'B1': done.
Results written to 1kgeas.B1.glm.firth .
End time: Tue Dec 27 23:04:40 2022
</code></pre>
<p>Lest check the first lines of the output:</p>
<pre><code>#CHROM  POS ID  REF ALT A1  TEST    OBS_CT  OR  LOG(OR)_SE  Z_STAT  P   ERRCODE
1   13273   1:13273:G:C G   C   C   ADD 503 0.746149    0.282904    -1.03509    0.300628    .
1   14599   1:14599:T:A T   A   A   ADD 503 1.67693 0.240899    2.14598 0.0318742.
1   14604   1:14604:A:G A   G   G   ADD 503 1.67693 0.240899    2.14598 0.0318742.
1   14930   1:14930:A:G A   G   G   ADD 503 1.64359 0.242872    2.04585 0.0407708.
1   69897   1:69897:T:C T   C   T   ADD 503 1.69142 0.200238    2.62471 0.00867216.
1   86331   1:86331:A:G A   G   G   ADD 503 1.41887 0.238055    1.46968 0.141649    .
1   91581   1:91581:G:A G   A   A   ADD 503 0.931304    0.123644    -0.5755980.564887   .
1   122872  1:122872:T:G    T   G   G   ADD 503 1.04828 0.182036    0.259034    0.795609    .
1   135163  1:135163:C:T    C   T   T   ADD 503 0.676666    0.242611    -1.60989    0.107422    .
</code></pre>
<h2 id="genomic-control">Genomic control</h2>
<p>Genomic control (GC) is a basic method for controlling for the confunding factors like population stratification.</p>
<p>We will calculate genomic control factor (lambda GC) to evaluate the inflation. The genomic control factor is calculated by dividing the <strong>median of observed Chi square statistics</strong> by the <strong>median of Chi square distribution with degree of 1</strong> (which is approximately 0.455).</p>
<p>$$ \lambda_{GC} = {median(\chi^{2}_{observed}) \over median(\chi^{2}_1)} $$</p>
<p>Then, we can used the genomic control factor to correct observed Chi suqare statistics.</p>
<p>$$ \chi^{2}<em>{corrected} = {\chi^{2}</em>{observed} \over \lambda_{GC}} $$</p>
<p>Reference: Devlin, B., &amp; Roeder, K. (1999). Genomic control for association studies. Biometrics, 55(4), 997-1004.</p>
<h2 id="significant-loci">Significant loci</h2>
<p>Please check <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<p>Loci that reached suggestive significance threhold (P value &lt; 5e-6) :</p>
<pre><code>SNPID   CHR POS EA  NEA SE  Z   P   OR  N   STATUS
1:217437563:C:T 1   217437563   C   T   0.151157    -5.22793    1.714210e-07    0.453736    503 9999999
2:55574452:G:C  2   55574452    C   G   0.160948    -5.98392    2.178320e-09    0.381707    503 9999999
3:176524872:C:T 3   176524872   T   C   0.248418    4.92774 8.318440e-07    3.401240    503 9999999
3:193128900:G:A 3   193128900   A   G   0.153788    4.70811 2.500290e-06    2.062770    503 9999999
6:29919659:T:C  6   29919659    T   C   0.155457    -5.89341    3.782970e-09    0.400048    503 9999999
9:36660672:A:G  9   36660672    G   A   0.160275    5.63422 1.758540e-08    2.467060    503 9999999
11:56249438:A:G 11  56249438    G   A   0.188891    -4.77836    1.767350e-06    0.405518    503 9999999
</code></pre>
<h1 id="visualization">Visualization</h1>
<p>To visualize the results, we will create manhattan plot, QQ plot and regional plot.</p>
<p>Please check <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<h2 id="mahanttan-plot-and-quantile-quantile-plot">Mahanttan plot and Quantile-Quantile plot</h2>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/209681591-dc691764-7346-4936-80b4-528bc425a61e.png" /></p>
<h2 id="regional-plot">Regional plot</h2>
<p>The regional plot for the loci of 2:55574452:G:C. </p>
<p>Please check <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/209681608-3973c546-ad52-4d77-a3a1-a1c60a7a0a97.png" /></p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>GWASTutorial is licensed under the MIT license</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>

        <div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
