
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="HE Yunye">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-8.5.7">
    
    
      
        <title>Association tests by PLINK - GWASTutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.20d9efc8.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
          
          
          <meta name="theme-color" content="#2094f3">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#association-test" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GWASTutorial" class="md-header__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GWASTutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Association tests by PLINK
            
          </span>
        </div>
      </div>
    </div>
    
    
    
    
      <div class="md-header__source">
        <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../02_Linux_basics/" class="md-tabs__link">
        Command Line
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../01_Dataset/" class="md-tabs__link">
        Pre-GWAS
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="./" class="md-tabs__link md-tabs__link--active">
        GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../07_Annotation/" class="md-tabs__link">
        Post-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../90_Recommended_Reading/" class="md-tabs__link">
        Others
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GWASTutorial" class="md-nav__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    GWASTutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_1">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Home" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Tutorial guide
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../99_About/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Command Line
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Command Line" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Command Line
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_1" type="checkbox" id="__nav_2_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_1">
          Linux
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Linux" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_1">
          <span class="md-nav__icon md-icon"></span>
          Linux
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../02_Linux_basics/" class="md-nav__link">
        Linux Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../82_windows_linux_subsystem/" class="md-nav__link">
        Windows Linux Subsystem
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../84_ssh/" class="md-nav__link">
        SSH
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../85_job_scheduler/" class="md-nav__link">
        Job scheduler
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../60_awk/" class="md-nav__link">
        Text editing - awk
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../61_sed/" class="md-nav__link">
        Text editing - sed
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../69_resources/" class="md-nav__link">
        Linux Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_2">
          Python
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Python" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          Python
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../70_python_basics/" class="md-nav__link">
        Python Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../71_python_resources/" class="md-nav__link">
        Python Resources
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../80_anaconda/" class="md-nav__link">
        Anaconda
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../81_jupyter_notebook/" class="md-nav__link">
        Jupyter notebook
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_3" type="checkbox" id="__nav_2_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_3">
          R
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="R" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_3">
          <span class="md-nav__icon md-icon"></span>
          R
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../75_R_basics/" class="md-nav__link">
        R Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../76_R_resources/" class="md-nav__link">
        R Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_4" type="checkbox" id="__nav_2_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_4">
          Other tools
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Other tools" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_4">
          <span class="md-nav__icon md-icon"></span>
          Other tools
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../83_git_and_github/" class="md-nav__link">
        Git and github
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Pre-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Pre-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../01_Dataset/" class="md-nav__link">
        Sample Dataset
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../03_Data_formats/" class="md-nav__link">
        Data Formats
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../04_Data_QC/" class="md-nav__link">
        Genotype Data QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../05_PCA/" class="md-nav__link">
        Sample PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../plot_PCA/" class="md-nav__link">
        Plotting PCA
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Association tests by PLINK
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Association tests by PLINK
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#file-preparation" class="md-nav__link">
    File preparation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genetic-models" class="md-nav__link">
    Genetic models
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#association-testing" class="md-nav__link">
    Association testing
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genomic-control" class="md-nav__link">
    Genomic control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#significant-loci" class="md-nav__link">
    Significant loci
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav" aria-label="Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mahanttan-plot" class="md-nav__link">
    Mahanttan plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantile-quantile-plot" class="md-nav__link">
    Quantile-quantile plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#regional-plot" class="md-nav__link">
    Regional plot
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Visualization/" class="md-nav__link">
        Visualization by gwaslab
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../33_linear_mixed_model/" class="md-nav__link">
        Linear mixed model (LMM)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../32_whole_genome_regression/" class="md-nav__link">
        Whole genome regression by REGENIE
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Post-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Post-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Post-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../07_Annotation/" class="md-nav__link">
        Annotation by ANNOVAR & VEP
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../15_winners_curse/" class="md-nav__link">
        Winner's curse
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../13_heritability/" class="md-nav__link">
        Heritability Concepts
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../14_gcta_greml/" class="md-nav__link">
        SNP-heritability estimation by GREML
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../08_LDSC/" class="md-nav__link">
        LD socre regression by LDSC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../09_Gene_based_analysis/" class="md-nav__link">
        Gene/Gene-set tests by MAGMA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../12_fine_mapping/" class="md-nav__link">
        Fine-mapping by susieR
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../11_meta_analysis/" class="md-nav__link">
        Meta-analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../10_PRS/" class="md-nav__link">
        Polygenic Risk Score
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../16_mendelian_randomization/" class="md-nav__link">
        Mendelian randomization
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../18_Conditioning_analysis/" class="md-nav__link">
        Conditional analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../17_colocalization/" class="md-nav__link">
        Colocalization
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Others
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Others" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Others
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../90_Recommended_Reading/" class="md-nav__link">
        Recommended reading
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../95_Assignment/" class="md-nav__link">
        Assignment - 1KG PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../96_Assignment2/" class="md-nav__link">
        Assignment - Presentation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#file-preparation" class="md-nav__link">
    File preparation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genetic-models" class="md-nav__link">
    Genetic models
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#association-testing" class="md-nav__link">
    Association testing
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genomic-control" class="md-nav__link">
    Genomic control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#significant-loci" class="md-nav__link">
    Significant loci
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav" aria-label="Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mahanttan-plot" class="md-nav__link">
    Mahanttan plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantile-quantile-plot" class="md-nav__link">
    Quantile-quantile plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#regional-plot" class="md-nav__link">
    Regional plot
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="association-test">Association test</h1>
<h2 id="file-preparation">File preparation</h2>
<p>To perform genome-wide association study (GWAS), usually we need the following files:</p>
<ul>
<li><strong>Genotype file</strong> (or dosage file) : usually in PLINK format, VCF format, or BGEN format.</li>
<li><strong>Phenotype file</strong> : plain text file.</li>
<li><strong>Covariate file</strong> (optional): plain text file. Usually covaraites include age, sex, and top Pricipal Components. </li>
</ul>
<p>For example,</p>
<div class="highlight"><pre><span></span><code># Phenotype file for a simulated binary trait; B1 is the phenotype name; 1 means control, 2 means case.
head 1kgeas_binary.txt 
FID IID B1
0 HG00403 2 
0 HG00404 1 
0 HG00406 2 
0 HG00407 2 
0 HG00409 2 
0 HG00410 2 
0 HG00419 1 
0 HG00421 2 
0 HG00422 1

# Covariate file (only top PCs calculated in the previous PCA section)
#FID    IID ALLELE_CT   NAMED_ALLELE_DOSAGE_SUM PC1_AVG PC2_AVG PC3_AVG PC4_AVG PC5_AVG PC6_AVG PC7_AVG PC8_AVG PC9_AVG PC10_AVG
0   HG00403 224016  224016  0.000246109 0.0292717   -0.0127437  -0.0135105  0.0301317   0.0196699   0.0232392   -0.0205941  -0.00416543 0.0121819
0   HG00404 224016  224016  -0.000716664    0.032043    -0.00573731 -0.0189504  0.0277385   -0.0154478  -0.0136551  -0.00147269 0.00510851  0.0268694
0   HG00406 224016  224016  0.00681476  0.0346759   -0.00706221 -0.0054266  -0.025458   -0.0166505  -0.00510707 -0.00105581 -0.0224028  0.0202523
0   HG00407 224016  224016  0.00695106  0.0244759   -0.00890072 -0.000694057    0.00946838  -0.00773378 -0.0139923  -0.0204871  -0.0003338  0.0387022
0   HG00409 224016  224016  -0.0023481  0.0213108   0.03235 0.0158793   0.026655    0.00324455-0.0107152    0.0317714   -0.00764455 0.0155973
0   HG00410 224016  224016  0.000926147 0.0198139   0.0475798   0.00314974  0.0275373   0.041886    -0.0133896  0.0184717   -0.0143644  0.0291036
0   HG00419 224016  224016  0.00580767  0.0369208   -0.00907507 0.00903163  -0.00649345 -0.000472359    -0.00327011 0.0160456   -0.005133   0.0141021
0   HG00421 224016  224016  0.000987901 0.0336895   -0.00697814 0.00557199  -0.0210537  0.00700968  0.00319921  -0.0215999  0.00127686  0.0350116
0   HG00422 224016  224016  0.00440288  0.0335901   -0.0125043  0.0135621   -0.0228428  0.00492741  0.00445856  -0.00911147 -0.00312742 -0.00784459
</code></pre></div>
<h2 id="genetic-models">Genetic models</h2>
<p>To test the association between a phenotype and genotypes, we also need to group the genotypes based on genetic models.</p>
<p>There are three basic genetic models:</p>
<ul>
<li>Additive model (ADD)</li>
<li>Dominant model (DOM)</li>
<li>Recessive model (REC)</li>
</ul>
<p>For example, suppose we have a biallelic SNP whose reference allele is A and alternative allele is G.</p>
<p>There are three possible genotypes for this SNP: AA, AG, and GG.</p>
<p>The table shows how we group different genotypes under each genetic model.</p>
<table>
<thead>
<tr>
<th>Genetic models</th>
<th>AA</th>
<th>AG</th>
<th>GG</th>
</tr>
</thead>
<tbody>
<tr>
<td>Additive model</td>
<td>0</td>
<td>1</td>
<td>2</td>
</tr>
<tr>
<td>Dominant model</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr>
<td>Recessive model</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>
</tbody>
</table>
<h2 id="association-testing">Association testing</h2>
<p>Please check https://www.cog-genomics.org/plink/2.0/assoc for more details.</p>
<p>For quantitative traits, we can employ a simple linear regression model to test associations:</p>
<div class="arithmatex">\[ 
y = G\beta_G + X\beta_X + e
\]</div>
<ul>
<li><span class="arithmatex">\(G\)</span> is the genotype matrix.</li>
<li><span class="arithmatex">\(\beta_G\)</span> is the effect sizes for variants.</li>
<li><span class="arithmatex">\(X\)</span> and <span class="arithmatex">\(\beta_X\)</span> are covariates and their effects.</li>
<li><span class="arithmatex">\(e\)</span> is the error term.</li>
</ul>
<p>For binary traits, we can utilize the logistic regression model to test associations:</p>
<div class="arithmatex">\[ 
logit(p) = G\beta_G + X\beta_X + e
\]</div>
<p>We will perform a logistic regression with firth correction for a simulated binary trait under additive model using the 1KG East Asian individuals.</p>
<div class="admonition note">
<p class="admonition-title">firth correction</p>
<p>Adding a penalty term to the log-likelihood function when fitting the logistic model, which results in less bias. - Firth, David. "Bias reduction of maximum likelihood estimates." Biometrika 80.1 (1993): 27-38.</p>
</div>
<p>Sample code:</p>
<div class="highlight"><pre><span></span><code>genotypeFile=&quot;../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot;  #!please set this to your own path
phenotypeFile=&quot;../01_Dataset/1kgeas_binary.txt&quot; #!please set this to your own path
covariateFile=&quot;../05_PCA/plink_results_projected.sscore&quot;

covariateCols=6-10
colName=&quot;B1&quot;
threadnum=2

plink2 \
    --bfile ${genotypeFile} \
    --pheno ${phenotypeFile} \
    --pheno-name ${colName} \
    --maf 0.01 \
    --covar ${covariateFile} \
    --covar-col-nums ${covariateCols} \
    --glm hide-covar firth \
    --threads ${threadnum} \
    --out 1kgeas
</code></pre></div>
<p>You will see a similar log like:</p>
<div class="highlight"><pre><span></span><code>PLINK v2.00a3LM 64-bit Intel (12 Dec 2020)     www.cog-genomics.org/plink/2.0/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to 1kgeas.log.
Options in effect:
  --bfile ../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020
  --covar ../05_PCA/plink_results_projected.sscore
  --covar-col-nums 6-10
  --glm hide-covar firth
  --maf 0.01
  --out 1kgeas
  --pheno ../01_Dataset/1kgeas_binary.txt
  --pheno-name B1
  --threads 2

Start time: Tue Dec 27 23:02:52 2022
15957 MiB RAM detected; reserving 7978 MiB for main workspace.
Using up to 2 compute threads.
504 samples (0 females, 0 males, 504 ambiguous; 504 founders) loaded from
../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam.
1122299 variants loaded from
../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim.
1 binary phenotype loaded (201 cases, 302 controls).
5 covariates loaded from ../05_PCA/plink_results_projected.sscore.
Calculating allele frequencies... done.
0 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
1122299 variants remaining after main filters.
--glm Firth regression on phenotype &#39;B1&#39;: done.
Results written to 1kgeas.B1.glm.firth .
End time: Tue Dec 27 23:04:40 2022
</code></pre></div>
<p>Lest check the first lines of the output:
<div class="highlight"><pre><span></span><code>#CHROM  POS ID  REF ALT A1  TEST    OBS_CT  OR  LOG(OR)_SE  Z_STAT  P   ERRCODE
1   13273   1:13273:G:C G   C   C   ADD 503 0.746149    0.282904    -1.03509    0.300628    .
1   14599   1:14599:T:A T   A   A   ADD 503 1.67693 0.240899    2.14598 0.0318742.
1   14604   1:14604:A:G A   G   G   ADD 503 1.67693 0.240899    2.14598 0.0318742.
1   14930   1:14930:A:G A   G   G   ADD 503 1.64359 0.242872    2.04585 0.0407708.
1   69897   1:69897:T:C T   C   T   ADD 503 1.69142 0.200238    2.62471 0.00867216.
1   86331   1:86331:A:G A   G   G   ADD 503 1.41887 0.238055    1.46968 0.141649    .
1   91581   1:91581:G:A G   A   A   ADD 503 0.931304    0.123644    -0.5755980.564887   .
1   122872  1:122872:T:G    T   G   G   ADD 503 1.04828 0.182036    0.259034    0.795609    .
1   135163  1:135163:C:T    C   T   T   ADD 503 0.676666    0.242611    -1.60989    0.107422    .
</code></pre></div></p>
<h2 id="genomic-control">Genomic control</h2>
<p>Genomic control (GC) is a basic method for controlling for the confunding factors like population stratification.</p>
<p>We will calculate genomic control factor (lambda GC) to evaluate the inflation. The genomic control factor is calculated by dividing the <strong>median of observed Chi square statistics</strong> by the <strong>median of Chi square distribution with degree of 1</strong> (which is approximately 0.455).</p>
<div class="arithmatex">\[ 
\lambda_{GC} = {median(\chi^{2}_{observed}) \over median(\chi^{2}_1)} 
\]</div>
<p>Then, we can used the genomic control factor to correct observed Chi suqare statistics.</p>
<div class="arithmatex">\[ 
\chi^{2}_{corrected} = {\chi^{2}_{observed} \over \lambda_{GC}} 
\]</div>
<p>Reference: Devlin, B., &amp; Roeder, K. (1999). Genomic control for association studies. Biometrics, 55(4), 997-1004.</p>
<h2 id="significant-loci">Significant loci</h2>
<p>Please check <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<p>Loci that reached suggestive significance threhold (P value &lt; 5e-6) :
<div class="highlight"><pre><span></span><code>SNPID   CHR POS EA  NEA SE  Z   P   OR  N   STATUS
1:217437563:C:T 1   217437563   C   T   0.151157    -5.22793    1.714210e-07    0.453736    503 9999999
2:55574452:G:C  2   55574452    C   G   0.160948    -5.98392    2.178320e-09    0.381707    503 9999999
3:176524872:C:T 3   176524872   T   C   0.248418    4.92774 8.318440e-07    3.401240    503 9999999
3:193128900:G:A 3   193128900   A   G   0.153788    4.70811 2.500290e-06    2.062770    503 9999999
6:29919659:T:C  6   29919659    T   C   0.155457    -5.89341    3.782970e-09    0.400048    503 9999999
9:36660672:A:G  9   36660672    G   A   0.160275    5.63422 1.758540e-08    2.467060    503 9999999
11:56249438:A:G 11  56249438    G   A   0.188891    -4.77836    1.767350e-06    0.405518    503 9999999
</code></pre></div></p>
<h2 id="visualization">Visualization</h2>
<p>To visualize the sumstats, we will create manhattan plot, QQ plot and regional plot.</p>
<p>Please check for codes : <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/209681591-dc691764-7346-4936-80b4-528bc425a61e.png" /></p>
<h3 id="mahanttan-plot">Mahanttan plot</h3>
<p>Mahanttan plot is the most classic visualization of GWAS summary statistics. It is a form of scatter plot. Each dot represents the test result for a variants. variants are sorted by its genome coordinates and are aligned along the X axis. Y axis shows the -log10(P value) for tests of variants in GWAS. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This kind of plot was named after Mahanttan in New York City since it resembles the Manhattan skyline.   </p>
</div>
<p><img width="686" alt="image" src="https://user-images.githubusercontent.com/40289485/209780549-54a24fdd-485b-4875-8f40-d6812eb644fe.png"></p>
<p>Data we need from sumstats to create mahattan plot:</p>
<ul>
<li>Chromosome </li>
<li>Basepair position</li>
<li>P value or -log10(P)</li>
</ul>
<div class="admonition tips">
<p class="admonition-title">Steps to create Mahattan plot</p>
<ol>
<li>sort the variants by genome coordinates.</li>
<li>map the genome coordinates of variants to the x axis.</li>
<li>convert P value to -log10(P).</li>
<li>create the scatter plot.</li>
</ol>
</div>
<h3 id="quantile-quantile-plot">Quantile-quantile plot</h3>
<p>Quantile-quantile plot (as known as Q-Q plot), is commonly used to compare an observed distribution with its expected distribution. For a specific point (x,y) on Q-Q plot, its y coordinate corresponds to one of the quantiles of the observed distribution, while its x coordinate corresponds to the same quantile of the expected distribution.</p>
<p>Quantile-quantile plot is used to check if there is any significant inflation in P value distribution, which usually indicates population stratification or cryptic relatedness. </p>
<p>Data we need from sumstats to create mahattan plot:</p>
<ul>
<li>P value or -log10(P)</li>
</ul>
<div class="admonition tips">
<p class="admonition-title">Steps to create Q-Q plot</p>
<p>Suppose we have <code>n</code> variants in our sumstats,</p>
<ol>
<li>convert the <code>n</code> P value to -log10(P).</li>
<li>sort the -log10(P) values in asending order.</li>
<li>get <code>n</code> numbers from <code>(0,1)</code> with equal intervals.</li>
<li>convert the <code>n</code> numbers to -log10(P) and sort in asending order.</li>
<li>create scatter plot using the sorted -log10(P) of sumstats as Y and sorted -log10(P) we generated as X.</li>
</ol>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The expected distribution of P value is an Uniform distribution from 0 to 1.</p>
</div>
<h3 id="regional-plot">Regional plot</h3>
<p>Mahanttan plot is very useful to check the overview of our sumstats. But is we want to check a specific genomic loci, we need a plot with finer resolution. This kind of plot is called regional plot. It is basically the Mahanttan plot of only a small region on genome, with points colored by its LD r2 with the lead variant in this region. </p>
<p>The regional plot for the loci of 2:55574452:G:C. </p>
<p>Please check <a href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab</a></p>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/209681608-3973c546-ad52-4d77-a3a1-a1c60a7a0a97.png" /></p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../plot_PCA/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Plotting PCA" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Plotting PCA
            </div>
          </div>
        </a>
      
      
        
        <a href="../Visualization/" class="md-footer__link md-footer__link--next" aria-label="Next: Visualization by gwaslab" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Visualization by gwaslab
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      GWASTutorial is licensed under the MIT license
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.top", "navigation.tracking", "navigation.expand", "navigation.path"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.8492ddcf.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>