
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="HE Yunye">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-8.5.7">
    
    
      
        <title>PRS - GWASTutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.20d9efc8.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
          
          
          <meta name="theme-color" content="#2094f3">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#polygenic-risk-scores" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GWASTutorial" class="md-header__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GWASTutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              PRS
            
          </span>
        </div>
      </div>
    </div>
    
    
    
    
      <div class="md-header__source">
        <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../02_Linux_basics/" class="md-tabs__link">
        Command Line
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../01_Dataset/" class="md-tabs__link">
        Pre-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../06_Association_tests/" class="md-tabs__link">
        GWAS
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../07_Annotation/" class="md-tabs__link md-tabs__link--active">
        Post-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../90_Recommended_Reading/" class="md-tabs__link">
        Others
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GWASTutorial" class="md-nav__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    GWASTutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_1">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Home" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Tutorial guide
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../99_About/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Command Line
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Command Line" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Command Line
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_1" type="checkbox" id="__nav_2_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_1">
          Linux
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Linux" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_1">
          <span class="md-nav__icon md-icon"></span>
          Linux
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../02_Linux_basics/" class="md-nav__link">
        Linux Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../82_windows_linux_subsystem/" class="md-nav__link">
        Windows Linux Subsystem
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../84_ssh/" class="md-nav__link">
        SSH
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../85_job_scheduler/" class="md-nav__link">
        Job scheduler
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../60_awk/" class="md-nav__link">
        Text editing - awk
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../61_sed/" class="md-nav__link">
        Text editing - sed
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../69_resources/" class="md-nav__link">
        Linux Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_2">
          Python
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Python" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          Python
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../70_python_basics/" class="md-nav__link">
        Python Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../71_python_resources/" class="md-nav__link">
        Python Resources
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../80_anaconda/" class="md-nav__link">
        Anaconda
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../81_jupyter_notebook/" class="md-nav__link">
        Jupyter notebook
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_3" type="checkbox" id="__nav_2_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2_3">
          Other tools
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Other tools" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_3">
          <span class="md-nav__icon md-icon"></span>
          Other tools
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../83_git_and_github/" class="md-nav__link">
        Git and github
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Pre-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Pre-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../01_Dataset/" class="md-nav__link">
        Sample Dataset
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../03_Data_formats/" class="md-nav__link">
        Data Formats
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../04_Data_QC/" class="md-nav__link">
        Genotype Data QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../05_PCA/" class="md-nav__link">
        Sample PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../plot_PCA/" class="md-nav__link">
        Plotting PCA
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../06_Association_tests/" class="md-nav__link">
        Association tests by PLINK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Visualization/" class="md-nav__link">
        Visualization by gwaslab
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../33_linear_mixed_model/" class="md-nav__link">
        Linear mixed model (LMM)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../32_whole_genome_regression/" class="md-nav__link">
        Whole genome regression by REGENIE
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Post-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Post-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Post-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../07_Annotation/" class="md-nav__link">
        Annotation by ANNOVAR & VEP
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../15_winners_curse/" class="md-nav__link">
        Winner's curse
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../13_heritability/" class="md-nav__link">
        Heritability Concepts
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../14_gcta_greml/" class="md-nav__link">
        SNP-heritability estimation by GREML
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../08_LDSC/" class="md-nav__link">
        LD socre regression by LDSC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../09_Gene_based_analysis/" class="md-nav__link">
        Gene/Gene-set tests by MAGMA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../12_fine_mapping/" class="md-nav__link">
        Fine-mapping by susieR
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          PRS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        PRS
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#definition" class="md-nav__link">
    Definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prs-analysis-workflow" class="md-nav__link">
    PRS Analysis Workflow
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methods" class="md-nav__link">
    Methods
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ctpt-using-plink" class="md-nav__link">
    C+T/P+T using PLINK
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-shrinkage-using-prs-cs" class="md-nav__link">
    Beta shrinkage using PRS-CS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameter-tuning" class="md-nav__link">
    Parameter tuning
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pgs-catalog" class="md-nav__link">
    PGS Catalog
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculate-prs-using-plink" class="md-nav__link">
    Calculate PRS using PLINK
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#regressions-for-evaluation-of-prs" class="md-nav__link">
    Regressions for evaluation of PRS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation" class="md-nav__link">
    Evaluation
  </a>
  
    <nav class="md-nav" aria-label="Evaluation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#roc-aic-auc-and-c-index" class="md-nav__link">
    ROC, AIC, AUC, and C-index
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r2-and-pseudo-r2" class="md-nav__link">
    R2 and pseudo-R2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r2-on-the-liability-scale-lee" class="md-nav__link">
    R2 on the liability scale (Lee)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#meta-grs" class="md-nav__link">
    Meta-GRS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Others
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Others" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Others
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../90_Recommended_Reading/" class="md-nav__link">
        Recommended reading
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../95_Assignment/" class="md-nav__link">
        Assignment - 1KG PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../96_Assignment2/" class="md-nav__link">
        Assignment - Presentation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#definition" class="md-nav__link">
    Definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prs-analysis-workflow" class="md-nav__link">
    PRS Analysis Workflow
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methods" class="md-nav__link">
    Methods
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ctpt-using-plink" class="md-nav__link">
    C+T/P+T using PLINK
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-shrinkage-using-prs-cs" class="md-nav__link">
    Beta shrinkage using PRS-CS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameter-tuning" class="md-nav__link">
    Parameter tuning
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pgs-catalog" class="md-nav__link">
    PGS Catalog
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculate-prs-using-plink" class="md-nav__link">
    Calculate PRS using PLINK
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#regressions-for-evaluation-of-prs" class="md-nav__link">
    Regressions for evaluation of PRS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation" class="md-nav__link">
    Evaluation
  </a>
  
    <nav class="md-nav" aria-label="Evaluation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#roc-aic-auc-and-c-index" class="md-nav__link">
    ROC, AIC, AUC, and C-index
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r2-and-pseudo-r2" class="md-nav__link">
    R2 and pseudo-R2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r2-on-the-liability-scale-lee" class="md-nav__link">
    R2 on the liability scale (Lee)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#meta-grs" class="md-nav__link">
    Meta-GRS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="polygenic-risk-scores">Polygenic risk scores</h1>
<h2 id="definition">Definition</h2>
<p>To calculate the PRS for sample j, </p>
<div class="arithmatex">\[PRS_j = \sum_{i=0}^{i=M} x_{i,j} \beta_{i}\]</div>
<ul>
<li><span class="arithmatex">\(\beta_i\)</span> : effect size for variant <span class="arithmatex">\(i\)</span></li>
<li><span class="arithmatex">\(x_{i,j}\)</span> : the effect allele count for sample <span class="arithmatex">\(j\)</span> at variant <span class="arithmatex">\(i\)</span></li>
<li><span class="arithmatex">\(M\)</span> : the number of variants</li>
</ul>
<h2 id="prs-analysis-workflow">PRS Analysis Workflow</h2>
<ol>
<li><strong>Developing PRS model</strong> using base data</li>
<li>Performing <strong>validation</strong> to obtain best-fit parameters</li>
<li><strong>Testing</strong> in an independent population</li>
</ol>
<h2 id="methods">Methods</h2>
<table>
<thead>
<tr>
<th>Category</th>
<th>Description</th>
<th>Representative Methods</th>
</tr>
</thead>
<tbody>
<tr>
<td>P value thresholding</td>
<td>P + T</td>
<td>C+T, PRSice</td>
</tr>
<tr>
<td>Beta shrinkage</td>
<td>genome-wide PRS model</td>
<td>LDpred, PRS-CS</td>
</tr>
</tbody>
</table>
<p>In this tutorial, we will first briefly introduce how to develop PRS model using the sample data and then demonstrate how we can download PRS models from PGS Catalog and apply to our sample genotype data. </p>
<h2 id="ctpt-using-plink">C+T/P+T using PLINK</h2>
<p>P+T stands for Pruning + Thresholding, also known as Clumping and Thresholding(C+T), which is a very simple and straightforward approach to construct PRS model.</p>
<div class="admonition info">
<p class="admonition-title">Clumping</p>
<p>Clumping: LD-pruning based on P value. It is a approach to select variants when there are multiple significant associations in high LD in the same region.</p>
<p>The three important parameters for clumping in PLINK are:</p>
<ul>
<li>clump-p1 0.0001       # Significance threshold for index SNPs</li>
<li>clump-r2 0.50         # LD threshold for clumping</li>
<li>clump-kb 250          # Physical distance threshold for clumping</li>
</ul>
</div>
<div class="admonition example">
<p class="admonition-title">Clumping using PLINK</p>
<div class="highlight"><pre><span></span><code>#!/bin/bash

plinkFile=../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020
sumStats=../06_Association_tests/1kgeas.B1.glm.firth

plink \
    --bfile ${plinkFile} \
    --clump-p1 0.0001 \
    --clump-r2 0.1 \
    --clump-kb 250 \
    --clump ${sumStats} \
    --clump-snp-field ID \
    --clump-field P \
    --out 1kg_eas
</code></pre></div>
<p>log
<div class="highlight"><pre><span></span><code>--clump: 40 clumps formed from 307 top variants.
</code></pre></div>
check only the header and the first "clump" of SNPs.</p>
<div class="highlight"><pre><span></span><code>head -n 2 1kg_eas.clumped
 CHR    F              SNP         BP        P    TOTAL   NSIG    S05    S01   S001  S0001    SP2
   2    1   2:55574452:G:C   55574452   2.18e-09      111     16     15     20     17     43 2:55376609:T:G(1),2:55376892:T:G(1),2:55379420:A:G(1),2:55384226:A:G(    1),2:55397522:A:T(1),2:55405037:T:A(1),2:55405468:G:A(1),2:55410835:C:T(1),2:55424229:G:A(1),2:55426447:A:G(1),2:55452014:C:T(1),2:55460751:C:T(1),2:55461053:A:G(    1),2:55465633:T:G(1),2:55470115:C:T(1),2:55478226:A:G(1),2:55482911:C:T(1),2:55489629:T:C(1),2:55491330:G:A(1),2:55495251:C:T(1),2:55500234:A:G(1),2:55501125:C:T(    1),2:55513738:C:T(1),2:55521039:A:G(1),2:55554956:C:A(1),2:55555726:T:C(1),2:55557192:C:T(1),2:55560347:C:A(1),2:55563020:A:G(1),2:55578761:C:T(1),2:55582635:T:C(    1),2:55584526:T:C(1),2:55585577:A:T(1),2:55587807:G:T(1),2:55598053:T:C(1),2:55602809:G:A(1),2:55605264:A:G(1),2:55607830:C:T(1),2:55609851:C:T(1),2:55610999:C:T(    1),2:55611741:A:G(1),2:55611766:T:C(1),2:55612986:G:C(1),2:55617255:C:T(1),2:55618813:C:T(1),2:55619407:C:A(1),2:55619923:C:T(1),2:55620927:G:A(1),2:55621660:T:C(    1),2:55622431:A:C(1),2:55625464:C:T(1),2:55632329:T:C(1),2:55632724:T:C(1),2:55635477:C:T(1),2:55636351:T:C(1),2:55636755:T:C(1),2:55636795:A:C(1),2:55638697:T:C(    1),2:55642350:T:C(1),2:55643047:G:A(1),2:55643666:A:C(1),2:55650512:G:A(1),2:55650572:G:C(1),2:55650886:G:A(1),2:55650940:C:A(1),2:55653053:G:C(1),2:55653269:C:A(    1),2:55654354:A:G(1),2:55654847:A:G(1),2:55659155:A:G(1),2:55662423:T:C(1),2:55669642:G:C(1),2:55685127:A:G(1),2:55685669:G:C(1),2:55687290:G:C(1),2:55695696:A:G(    1),2:55696124:A:G(1),2:55709416:G:A(1),2:55720739:G:C(1),2:55724035:T:C(1)
</code></pre></div>
</div>
<h2 id="beta-shrinkage-using-prs-cs">Beta shrinkage using PRS-CS</h2>
<div class="arithmatex">\[ \beta_j | \Phi_j \sim N(0,\phi\Phi_j) ,  \Phi_j \sim g \]</div>
<p>Reference: Ge, T., Chen, C. Y., Ni, Y., Feng, Y. C. A., &amp; Smoller, J. W. (2019). Polygenic prediction via Bayesian regression and continuous shrinkage priors. Nature communications, 10(1), 1-10.</p>
<h2 id="parameter-tuning">Parameter tuning</h2>
<p>|Method|Description|
|-|-|-|
|Cross-validation| 10-fold cross validation. This method usually requires large-scale genotype dataset.|
|Independent population| Perform validation in an independent population of the same ancestry. |
|Pseudo-validation|A few methods can estimate a single optimal shrinkage parameter using only the base GWAS summary statistics.|</p>
<h2 id="pgs-catalog">PGS Catalog</h2>
<p>Just like GWAS Catalog, you can now download published PRS  models from PGS catalog. </p>
<p>URL: http://www.pgscatalog.org/</p>
<h2 id="calculate-prs-using-plink">Calculate PRS using PLINK</h2>
<div class="highlight"><pre><span></span><code>plink --score
</code></pre></div>
<h2 id="regressions-for-evaluation-of-prs">Regressions for evaluation of PRS</h2>
<div class="arithmatex">\[Phenotype \sim PRS_{phenotype} + Covariates\]</div>
<div class="arithmatex">\[logit(P) \sim PRS_{phenotype} + Covariates\]</div>
<p>Covariates usually include sex, age and top 10 PCs.</p>
<h2 id="evaluation">Evaluation</h2>
<h3 id="roc-aic-auc-and-c-index">ROC, AIC, AUC, and C-index</h3>
<p>ROC : receiver operating characteristic curve shows the performance of a classification model at all thresholds.</p>
<ul>
<li>"y" : True Positive rate. <span class="arithmatex">\({{TP}\over{TP + FN}}\)</span></li>
<li>"x" : False Positive rate. <span class="arithmatex">\({{FP}\over{FP + TN}}\)</span></li>
</ul>
<p>AUC: area under the ROC Curve, a common measure for the performance of a classification model.</p>
<p>Akaike Information Criterion (AIC): a measure for comparison of different statistical models.</p>
<div class="arithmatex">\[AIC = 2k - 2ln(\hat{L})\]</div>
<ul>
<li><span class="arithmatex">\(k\)</span> : number of estimated parameters</li>
<li><span class="arithmatex">\(\hat{L}\)</span> : maximum value of the model likelihood function</li>
</ul>
<p><strong>C-index</strong>: concordance index, which is a metric to evaluate the predictive performance of models and commonly used in survival analysis. It is a measure of the probability that the predicted scores <span class="arithmatex">\(M_i\)</span> and $ M_j$ by a model of two randomly selected individuals <span class="arithmatex">\(i\)</span> and <span class="arithmatex">\(j\)</span>, have the reverse relative order as their true event times <span class="arithmatex">\(T_i, T_j\)</span>.</p>
<div class="arithmatex">\[ C = Pr (M_j &gt; M_i | T_j &lt; T_i) \]</div>
<p>Interpretation: Individuals with higher scores should have higher risk of the disease events</p>
<div class="admonition info">
<p class="admonition-title">C-index</p>
<p>Reference: Harrell, F. E., Califf, R. M., Pryor, D. B., Lee, K. L., &amp; Rosati, R. A. (1982). Evaluating the yield of medical tests. Jama, 247(18), 2543-2546.
Reference: Longato, E., Vettoretti, M., &amp; Di Camillo, B. (2020). A practical perspective on the concordance index for the evaluation and selection of prognostic time-to-event models. Journal of Biomedical Informatics, 108, 103496.</p>
</div>
<h3 id="r2-and-pseudo-r2">R2 and pseudo-R2</h3>
<div class="admonition info">
<p class="admonition-title">Coefficient of determination</p>
<p><span class="arithmatex">\(R^2\)</span> : coefficient of determination, which measures the amount of variance explained by the regression model.</p>
<p>In linear regression:</p>
<div class="arithmatex">\[ R^2 = 1 - {{RSS}\over{TSS}} \]</div>
<ul>
<li><span class="arithmatex">\(RSS\)</span> : sum of squares of residuals</li>
<li><span class="arithmatex">\(TSS\)</span> : total sum of squares</li>
</ul>
</div>
<div class="admonition info">
<p class="admonition-title">Pseudo-R2 (Nagelkerke)</p>
<p>In logistic regression, </p>
<p>One of the most commonly used Pseudo-R2 for PRS analysis is Nagelkerke's <span class="arithmatex">\(R^2\)</span></p>
<div class="arithmatex">\[R^2_{Nagelkerke} = {{1 - ({{L_0}\over{L_M}})^{2/n}}\over{1 - L_0^{2/n}}}\]</div>
<ul>
<li><span class="arithmatex">\(L_0\)</span> : Likelihood of the null model</li>
<li><span class="arithmatex">\(L_full\)</span> : Likelihood of the full model</li>
</ul>
</div>
<h3 id="r2-on-the-liability-scale-lee">R2 on the liability scale (Lee)</h3>
<div class="admonition info">
<p class="admonition-title">R2 on liability scale</p>
<p><span class="arithmatex">\(R^2\)</span> on the liability scale for ascertained case-control studies</p>
<div class="arithmatex">\[ R^2_l = {{R_o^2 C}\over{1 + R_o^2 \theta C }} \]</div>
<ul>
<li><span class="arithmatex">\(C\)</span> and <span class="arithmatex">\(\theta\)</span> are correcting factors for ascertainment</li>
<li><span class="arithmatex">\(C = {{K(1-K)}\over{Z^2}}{{K(1-K)}\over{P(1-P)}}\)</span> </li>
<li>
<p><span class="arithmatex">\(\theta = m {{P-K}\over{1-K}} ( m{{P-K}\over{1-K}} - t)\)</span>  </p>
</li>
<li>
<p><span class="arithmatex">\(K\)</span> : population disease prevalence</p>
</li>
<li><span class="arithmatex">\(P\)</span> : sample disease prevalence</li>
<li><span class="arithmatex">\(t\)</span>: the threshold on the normal distribution truncating the proportion of disease prevalence K</li>
<li><span class="arithmatex">\(m = z / K\)</span> : mean liability for cases</li>
<li><span class="arithmatex">\(z = f(t)\)</span> : the value of probability density function of standard normal distribution at t </li>
</ul>
<p>Reference : Lee, S. H., Goddard, M. E., Wray, N. R., &amp; Visscher, P. M. (2012). A better coefficient of determination for genetic profile analysis. Genetic epidemiology, 36(3), 214-224.</p>
<p>The authors also provided R codes for calculation (removed unrelated codes for simplicity)</p>
<div class="highlight"><pre><span></span><code><span class="c1"># R2 on the liability scale using the transformation</span>

<span class="n">nt</span> <span class="o">=</span> <span class="n">total</span> <span class="n">number</span> <span class="n">of</span> <span class="n">the</span> <span class="n">sample</span>
<span class="n">ncase</span> <span class="o">=</span> <span class="n">number</span> <span class="n">of</span> <span class="n">cases</span>
<span class="n">ncont</span> <span class="o">=</span> <span class="n">number</span> <span class="n">of</span> <span class="n">controls</span>
<span class="n">thd</span> <span class="o">=</span> <span class="n">the</span> <span class="n">threshold</span> <span class="n">on</span> <span class="n">the</span> <span class="n">normal</span> <span class="n">distribution</span> <span class="n">which</span> <span class="n">truncates</span> <span class="n">the</span> <span class="n">proportion</span> <span class="n">of</span> <span class="n">disease</span> <span class="n">prevalence</span>
<span class="n">K</span> <span class="o">=</span> <span class="n">population</span> <span class="n">prevalence</span>
<span class="n">P</span> <span class="o">=</span> <span class="n">proportion</span> <span class="n">of</span> <span class="n">cases</span> <span class="n">in</span> <span class="n">the</span> <span class="n">case</span><span class="o">-</span><span class="n">control</span> <span class="n">samples</span>

<span class="c1">#threshold</span>
<span class="n">thd</span> <span class="o">=</span> <span class="o">-</span><span class="nf">qnorm</span><span class="p">(</span><span class="n">K</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>

<span class="c1">#value of standard normal density function at thd</span>
<span class="n">zv</span> <span class="o">=</span> <span class="nf">dnorm</span><span class="p">(</span><span class="n">thd</span><span class="p">)</span> 

<span class="c1">#mean liability for case</span>
<span class="n">mv</span> <span class="o">=</span> <span class="n">zv</span><span class="o">/</span><span class="n">K</span> 

<span class="c1">#linear model</span>
<span class="n">lmv</span> <span class="o">=</span> <span class="nf">lm</span><span class="p">(</span><span class="n">y</span>∼<span class="n">g</span><span class="p">)</span> 

<span class="c1">#R20 : R2 on the observed scale</span>
<span class="n">R2O</span> <span class="o">=</span> <span class="nf">var</span><span class="p">(</span><span class="n">lmv</span><span class="o">$</span><span class="n">fitted.values</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="n">ncase</span><span class="o">/</span><span class="n">nt</span><span class="o">*</span><span class="n">ncont</span><span class="o">/</span><span class="n">nt</span><span class="p">)</span>

<span class="c1"># calculate correction factors</span>
<span class="n">theta</span> <span class="o">=</span> <span class="n">mv</span><span class="o">*</span><span class="p">(</span><span class="n">P</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">*</span><span class="p">(</span><span class="n">mv</span><span class="o">*</span><span class="p">(</span><span class="n">P</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">-</span><span class="n">thd</span><span class="p">)</span> 
<span class="n">cv</span> <span class="o">=</span> <span class="n">K</span><span class="o">*</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">/</span><span class="n">zv</span><span class="o">^</span><span class="m">2</span><span class="o">*</span><span class="n">K</span><span class="o">*</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">K</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="n">P</span><span class="o">*</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">P</span><span class="p">))</span> 

<span class="c1"># convert to R2 on the liability scale</span>
<span class="n">R2</span> <span class="o">=</span> <span class="n">R2O</span><span class="o">*</span><span class="n">cv</span><span class="o">/</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">R2O</span><span class="o">*</span><span class="n">theta</span><span class="o">*</span><span class="n">cv</span><span class="p">)</span>
</code></pre></div>
</div>
<h2 id="meta-grs">Meta-GRS</h2>
<h2 id="reference">Reference</h2>
<ul>
<li>PLINK : Purcell, Shaun, et al. "PLINK: a tool set for whole-genome association and population-based linkage analyses." The American journal of human genetics 81.3 (2007): 559-575.</li>
<li>PGS Catalog : Lambert, Samuel A., et al. "The Polygenic Score Catalog as an open database for reproducibility and systematic evaluation." Nature Genetics 53.4 (2021): 420-425.</li>
<li>PRS-CS: Ge, Tian, et al. "Polygenic prediction via Bayesian regression and continuous shrinkage priors." Nature communications 10.1 (2019): 1-10.</li>
<li>PRS Tutorial: Choi, Shing Wan, Timothy Shin-Heng Mak, and Paul F. O’Reilly. "Tutorial: a guide to performing polygenic risk score analyses." Nature protocols 15.9 (2020): 2759-2772.</li>
<li>Lee, S. H., Goddard, M. E., Wray, N. R., &amp; Visscher, P. M. (2012). A better coefficient of determination for genetic profile analysis. Genetic epidemiology, 36(3), 214-224.</li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../12_fine_mapping/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Fine-mapping by susieR" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Fine-mapping by susieR
            </div>
          </div>
        </a>
      
      
        
        <a href="../90_Recommended_Reading/" class="md-footer__link md-footer__link--next" aria-label="Next: Recommended reading" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Recommended reading
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      GWASTutorial is licensed under the MIT license
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.top", "navigation.tracking", "navigation.expand", "navigation.path"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.8492ddcf.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>