
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="HE Yunye">
      
      
      
        <link rel="prev" href="../prs_tutorial/">
      
      
        <link rel="next" href="../TwoSampleMR/">
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>MR Basics - GWASTutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#mendelian-randomization" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GWASTutorial" class="md-header__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GWASTutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              MR Basics
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../02_Linux_basics/" class="md-tabs__link">
        Command Line
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../01_Dataset/" class="md-tabs__link">
        Pre-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../06_Association_tests/" class="md-tabs__link">
        GWAS
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../07_Annotation/" class="md-tabs__link md-tabs__link--active">
        Post-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../19_ld/" class="md-tabs__link">
        Topics
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../95_Assignment/" class="md-tabs__link">
        Assignments
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GWASTutorial" class="md-nav__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    GWASTutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Tutorial guide
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../99_About/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Command Line
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Command Line
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1" >
      
      
      
        <label class="md-nav__link" for="__nav_2_1" id="__nav_2_1_label" tabindex="0">
          Linux
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_1">
          <span class="md-nav__icon md-icon"></span>
          Linux
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../02_Linux_basics/" class="md-nav__link">
        Linux Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../82_windows_linux_subsystem/" class="md-nav__link">
        Windows Linux Subsystem
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../84_ssh/" class="md-nav__link">
        SSH
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../85_job_scheduler/" class="md-nav__link">
        Job scheduler
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../60_awk/" class="md-nav__link">
        Text editing - awk
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../61_sed/" class="md-nav__link">
        Text editing - sed
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../69_resources/" class="md-nav__link">
        Linux Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
          Python
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          Python
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../70_python_basics/" class="md-nav__link">
        Python Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../71_python_resources/" class="md-nav__link">
        Python Resources
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../80_anaconda/" class="md-nav__link">
        Anaconda
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../81_jupyter_notebook/" class="md-nav__link">
        Jupyter notebook
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_3" >
      
      
      
        <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
          R
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_3">
          <span class="md-nav__icon md-icon"></span>
          R
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../75_R_basics/" class="md-nav__link">
        R Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../76_R_resources/" class="md-nav__link">
        R Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_4" >
      
      
      
        <label class="md-nav__link" for="__nav_2_4" id="__nav_2_4_label" tabindex="0">
          Other tools
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_4">
          <span class="md-nav__icon md-icon"></span>
          Other tools
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../83_git_and_github/" class="md-nav__link">
        Git and github
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Pre-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Pre-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../01_Dataset/" class="md-nav__link">
        Sample Dataset
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../03_Data_formats/" class="md-nav__link">
        Data Formats
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../04_Data_QC/" class="md-nav__link">
        Genotype Data QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../05_PCA/" class="md-nav__link">
        Sample PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../plot_PCA/" class="md-nav__link">
        Plotting PCA
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../06_Association_tests/" class="md-nav__link">
        Association tests by PLINK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Visualization/" class="md-nav__link">
        Visualization by gwaslab
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../33_linear_mixed_model/" class="md-nav__link">
        Linear mixed model (LMM)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../32_whole_genome_regression/" class="md-nav__link">
        Whole genome regression by REGENIE
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Post-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Post-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../07_Annotation/" class="md-nav__link">
        Annotation by ANNOVAR & VEP
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../14_gcta_greml/" class="md-nav__link">
        SNP-heritability estimation by GREML
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../08_LDSC/" class="md-nav__link">
        LD socre regression by LDSC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../09_Gene_based_analysis/" class="md-nav__link">
        Gene/Gene-set tests by MAGMA
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Fine-mapping by susieR
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Fine-mapping by susieR
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../12_fine_mapping/" class="md-nav__link">
        Fine-mapping Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../finemapping_susie/" class="md-nav__link">
        Finemapping using susieR
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../11_meta_analysis/" class="md-nav__link">
        Meta-analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_7" >
      
      
      
        <label class="md-nav__link" for="__nav_5_7" id="__nav_5_7_label" tabindex="0">
          Polygenic Risk Score
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_7_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_7">
          <span class="md-nav__icon md-icon"></span>
          Polygenic Risk Score
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../10_PRS/" class="md-nav__link">
        PRS Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../PRS_evaluation/" class="md-nav__link">
        PRS evaluation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../prs_tutorial/" class="md-nav__link">
        PRS Tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_8" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5_8" id="__nav_5_8_label" tabindex="0">
          Mendelian randomization
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_8_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_8">
          <span class="md-nav__icon md-icon"></span>
          Mendelian randomization
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          MR Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        MR Basics
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#mendelian-randomization-introduction" class="md-nav__link">
    Mendelian randomization introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#instrumental-variables-iv" class="md-nav__link">
    Instrumental Variables (IV)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assumptions" class="md-nav__link">
    Assumptions
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#two-stage-least-squares-2sls" class="md-nav__link">
    Two-stage least-squares (2SLS)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#two-sample-mr" class="md-nav__link">
    Two-sample MR
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iv-selection" class="md-nav__link">
    IV selection
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice" class="md-nav__link">
    Practice
  </a>
  
    <nav class="md-nav" aria-label="Practice">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#file-preparation" class="md-nav__link">
    File Preparation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r-package-twosamplemr" class="md-nav__link">
    R package TwoSampleMR
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#loading-package" class="md-nav__link">
    Loading package
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reading-exposure-sumstats" class="md-nav__link">
    Reading exposure sumstats
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extracting-instrumental-variables" class="md-nav__link">
    Extracting instrumental variables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clumping-exposure-variables" class="md-nav__link">
    Clumping exposure variables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outcome" class="md-nav__link">
    outcome
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#harmonizing-data" class="md-nav__link">
    Harmonizing data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-mr-analysis" class="md-nav__link">
    Perform MR analysis
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sensitivity-analysis" class="md-nav__link">
    Sensitivity analysis
  </a>
  
    <nav class="md-nav" aria-label="Sensitivity analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#heterogeneity" class="md-nav__link">
    Heterogeneity
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#horizontal-pleiotropy" class="md-nav__link">
    Horizontal Pleiotropy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-snp-mr-and-leave-one-out-mr" class="md-nav__link">
    Single SNP MR and leave-one-out MR
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav" aria-label="Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#scatter-plot" class="md-nav__link">
    Scatter plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-snp" class="md-nav__link">
    Single SNP
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#leave-one-out" class="md-nav__link">
    Leave-one-out
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#funnel-plot" class="md-nav__link">
    Funnel plot
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mr-steiger-directionality-test" class="md-nav__link">
    MR Steiger directionality test
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mr-base-web-app" class="md-nav__link">
    MR-Base (web app)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hemani-g-zheng-j-elsworth-b-wade-k-h-haberland-v-baird-d-haycock-p-c-2018-the-mr-base-platform-supports-systematic-causal-inference-across-the-human-phenome-elife-7-e34408" class="md-nav__link">
    Hemani, G., Zheng, J., Elsworth, B., Wade, K. H., Haberland, V., Baird, D., ... &amp; Haycock, P. C. (2018). The MR-Base platform supports systematic causal inference across the human phenome. elife, 7, e34408.
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../TwoSampleMR/" class="md-nav__link">
        TwoSampleMR Tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../18_Conditioning_analysis/" class="md-nav__link">
        Conditional analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../17_colocalization/" class="md-nav__link">
        Colocalization
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
      
      
      
        <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Topics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Topics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../19_ld/" class="md-nav__link">
        Linkage Disequilibrium (LD)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../13_heritability/" class="md-nav__link">
        Heritability Concepts
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../20_power_analysis/" class="md-nav__link">
        Statistical power
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../15_winners_curse/" class="md-nav__link">
        Winner's curse
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../90_Recommended_Reading/" class="md-nav__link">
        Recommended reading
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
      
      
      
        <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Assignments
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Assignments
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../95_Assignment/" class="md-nav__link">
        Assignment - 1KG PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../96_Assignment2/" class="md-nav__link">
        Assignment - Presentation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#mendelian-randomization-introduction" class="md-nav__link">
    Mendelian randomization introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#instrumental-variables-iv" class="md-nav__link">
    Instrumental Variables (IV)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assumptions" class="md-nav__link">
    Assumptions
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#two-stage-least-squares-2sls" class="md-nav__link">
    Two-stage least-squares (2SLS)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#two-sample-mr" class="md-nav__link">
    Two-sample MR
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iv-selection" class="md-nav__link">
    IV selection
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice" class="md-nav__link">
    Practice
  </a>
  
    <nav class="md-nav" aria-label="Practice">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#file-preparation" class="md-nav__link">
    File Preparation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r-package-twosamplemr" class="md-nav__link">
    R package TwoSampleMR
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#loading-package" class="md-nav__link">
    Loading package
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reading-exposure-sumstats" class="md-nav__link">
    Reading exposure sumstats
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extracting-instrumental-variables" class="md-nav__link">
    Extracting instrumental variables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clumping-exposure-variables" class="md-nav__link">
    Clumping exposure variables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outcome" class="md-nav__link">
    outcome
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#harmonizing-data" class="md-nav__link">
    Harmonizing data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-mr-analysis" class="md-nav__link">
    Perform MR analysis
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sensitivity-analysis" class="md-nav__link">
    Sensitivity analysis
  </a>
  
    <nav class="md-nav" aria-label="Sensitivity analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#heterogeneity" class="md-nav__link">
    Heterogeneity
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#horizontal-pleiotropy" class="md-nav__link">
    Horizontal Pleiotropy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-snp-mr-and-leave-one-out-mr" class="md-nav__link">
    Single SNP MR and leave-one-out MR
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav" aria-label="Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#scatter-plot" class="md-nav__link">
    Scatter plot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-snp" class="md-nav__link">
    Single SNP
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#leave-one-out" class="md-nav__link">
    Leave-one-out
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#funnel-plot" class="md-nav__link">
    Funnel plot
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mr-steiger-directionality-test" class="md-nav__link">
    MR Steiger directionality test
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mr-base-web-app" class="md-nav__link">
    MR-Base (web app)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hemani-g-zheng-j-elsworth-b-wade-k-h-haberland-v-baird-d-haycock-p-c-2018-the-mr-base-platform-supports-systematic-causal-inference-across-the-human-phenome-elife-7-e34408" class="md-nav__link">
    Hemani, G., Zheng, J., Elsworth, B., Wade, K. H., Haberland, V., Baird, D., ... &amp; Haycock, P. C. (2018). The MR-Base platform supports systematic causal inference across the human phenome. elife, 7, e34408.
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="mendelian-randomization">Mendelian randomization</h1>
<h2 id="mendelian-randomization-introduction">Mendelian randomization introduction</h2>
<div class="admonition tip">
<p class="admonition-title">Comparison between RCT and MR</p>
<p><img width="636" alt="image" src="https://user-images.githubusercontent.com/40289485/219572347-1ebc2f0a-3c9a-49a0-a058-638e4973873e.png"></p>
</div>
<h2 id="instrumental-variables-iv">Instrumental Variables (IV)</h2>
<p>Instrumental variable (IV) can be defined as a variable  that is correlated with the exposure X and uncorrelated with the error <span class="arithmatex">\(\epsilon\)</span> in the following regression: </p>
<div class="arithmatex">\[ Y = X\beta + \epsilon \]</div>
<ul>
<li><span class="arithmatex">\(Y\)</span> is the outcome</li>
</ul>
<h2 id="assumptions">Assumptions</h2>
<div class="admonition danger">
<p class="admonition-title">Key Assumptions</p>
<table>
<thead>
<tr>
<th>Assumptions</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Relevance</strong></td>
<td>Instrumental variables are strongly associated with the exposure.</td>
</tr>
<tr>
<td><strong>Exclusion restriction</strong></td>
<td>Instrumental variables do not affect the outcome except through the exposure.</td>
</tr>
<tr>
<td><strong>Independence</strong></td>
<td>There are no confounders of the instrumental variables and the outcome.</td>
</tr>
<tr>
<td>Monotonicity</td>
<td>Variants affect the exposure in the same direction for all individuals</td>
</tr>
<tr>
<td>No assortative mating</td>
<td>Assortative mating might cause bias in MR</td>
</tr>
</tbody>
</table>
</div>
<h2 id="two-stage-least-squares-2sls">Two-stage least-squares (2SLS)</h2>
<div class="arithmatex">\[ X = \mu_1 + \beta_{IV} IV + \epsilon_1  \]</div>
<div class="arithmatex">\[ Y = \mu_2 + \beta_{2SLS} \hat{X} + \epsilon_2 \]</div>
<h2 id="two-sample-mr">Two-sample MR</h2>
<p>Two-sample MR refers to the approach that the genetic effects of the instruments on the exposure can be estimated in an independent sample other than that used to estimate effects between instruments on the outcome. As more and more GWAS summary statistics become publicly available, the scope of MR also expands with Two-sample MR methods.</p>
<div class="arithmatex">\[ \hat{\beta}_{X,Y} = {{\hat{\beta}_{IV,Y}}\over{\hat{\beta}_{IV,X}}} \]</div>
<div class="admonition warning">
<p class="admonition-title">Caveats</p>
<p>For two-sample MR, there is an additional key assumption:</p>
<p><strong>The two samples used for MR are from the same underlying populations. (The effect size of instruments on exposure should be the same in both samples.)</strong> </p>
<p>Therefore, for two-sample MR, we usually use datasets from similar non-overlapping populations in terms of not only ancestry but also contextual factors. </p>
</div>
<h2 id="iv-selection">IV selection</h2>
<p>One of the first things to do when you plan to perform any type of MR is to check the associations of instrumental variables with the exposure to avoid bias caused by weak IVs.</p>
<p>The most commonly used method here is the <strong>F-statistic</strong>, which tests the association of instrumental variables with the exposure.</p>
<h2 id="practice">Practice</h2>
<p>In this tutorial, we will walk you through how to perform a minimal TwoSampleMR analysis. We will use the R package <a href="https://mrcieu.github.io/TwoSampleMR/index.html">TwoSampleMR</a>, which provides easy-to-use functions for formatting, clumping and harmonizing GWAS summary statistics. </p>
<p>This package integrates a variety of commonly used MR methods for analysis, including:
<div class="highlight"><pre><span></span><code>&gt; mr_method_list()
                             obj
1                  mr_wald_ratio
2               mr_two_sample_ml
3            mr_egger_regression
4  mr_egger_regression_bootstrap
5               mr_simple_median
6             mr_weighted_median
7   mr_penalised_weighted_median
8                         mr_ivw
9                  mr_ivw_radial
10                    mr_ivw_mre
11                     mr_ivw_fe
12                mr_simple_mode
13              mr_weighted_mode
14         mr_weighted_mode_nome
15           mr_simple_mode_nome
16                       mr_raps
17                       mr_sign
18                        mr_uwr

                                                        name PubmedID
1                                                 Wald ratio
2                                         Maximum likelihood
3                                                   MR Egger 26050253
4                                       MR Egger (bootstrap) 26050253
5                                              Simple median
6                                            Weighted median
7                                  Penalised weighted median
8                                  Inverse variance weighted
9                                                 IVW radial
10 Inverse variance weighted (multiplicative random effects)
11                 Inverse variance weighted (fixed effects)
12                                               Simple mode
13                                             Weighted mode
14                                      Weighted mode (NOME)
15                                        Simple mode (NOME)
16                      Robust adjusted profile score (RAPS)
17                                     Sign concordance test
18                                     Unweighted regression
</code></pre></div></p>
<h3 id="file-preparation">File Preparation</h3>
<p>To perform two-sample MR analysis, we need summary statistics for exposure and outcome generated from independent populations with the same ancestry.</p>
<p>In this tutorial, we will use sumstats from <a href="https://pheweb.jp/">Biobank Japan pheweb</a> and <a href="https://koges.leelabsg.org/">KoGES pheweb</a>.</p>
<ul>
<li>Type 2 diabetes sumstats from BBJ : <code>wget -O bbj_t2d.zip https://pheweb.jp/download/T2D</code></li>
<li>BMI sumstats from KoGES :  <code>wget -O koges_bmi.txt.gz https://koges.leelabsg.org/download/KoGES_BMI</code></li>
</ul>
<h3 id="r-package-twosamplemr">R package TwoSampleMR</h3>
<p>First, to use TwosampleMR, we need R&gt;= 4.1.
To install the package, run:</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">remotes</span><span class="p">)</span>
<span class="nf">install_github</span><span class="p">(</span><span class="s">&quot;MRCIEU/TwoSampleMR&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="loading-package">Loading package</h3>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">TwoSampleMR</span><span class="p">)</span>
</code></pre></div>
<h3 id="reading-exposure-sumstats">Reading exposure sumstats</h3>
<div class="highlight"><pre><span></span><code><span class="c1">#format exposures dataset</span>

<span class="n">exp_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;koges_bmi.txt.gz&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="extracting-instrumental-variables">Extracting instrumental variables</h3>
<div class="highlight"><pre><span></span><code><span class="c1"># select only significant variants</span>
<span class="n">exp_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">exp_raw</span><span class="p">,</span><span class="n">exp_raw</span><span class="o">$</span><span class="n">pval</span><span class="o">&lt;</span><span class="m">5e-8</span><span class="p">)</span>

<span class="n">exp_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">format_data</span><span class="p">(</span><span class="w"> </span><span class="n">exp_raw</span><span class="p">,</span>
<span class="w">    </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;exposure&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">snp_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;rsids&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">beta_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;beta&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">se_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sebeta&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">effect_allele_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;alt&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">other_allele_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ref&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">eaf_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;af&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">pval_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;pval&quot;</span><span class="p">,</span>
<span class="p">)</span>
</code></pre></div>
<h3 id="clumping-exposure-variables">Clumping exposure variables</h3>
<div class="highlight"><pre><span></span><code><span class="n">clumped_exp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">clump_data</span><span class="p">(</span><span class="n">exp_dat</span><span class="p">,</span><span class="n">clump_r2</span><span class="o">=</span><span class="m">0.01</span><span class="p">,</span><span class="n">pop</span><span class="o">=</span><span class="s">&quot;EAS&quot;</span><span class="p">)</span><span class="w"> </span>
</code></pre></div>
<h3 id="outcome">outcome</h3>
<div class="highlight"><pre><span></span><code><span class="n">out_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;hum0197.v3.BBJ.T2D.v1/GWASsummary_T2D_Japanese_SakaueKanai2020.auto.txt.gz&quot;</span><span class="p">,</span>
<span class="w">                    </span><span class="n">select</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;SNPID&quot;</span><span class="p">,</span><span class="s">&quot;Allele1&quot;</span><span class="p">,</span><span class="s">&quot;Allele2&quot;</span><span class="p">,</span><span class="s">&quot;BETA&quot;</span><span class="p">,</span><span class="s">&quot;SE&quot;</span><span class="p">,</span><span class="s">&quot;p.value&quot;</span><span class="p">))</span>
<span class="n">out_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">format_data</span><span class="p">(</span><span class="w"> </span><span class="n">out_raw</span><span class="p">,</span>
<span class="w">    </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;outcome&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">snp_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SNPID&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">beta_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BETA&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">se_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SE&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">effect_allele_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Allele2&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">other_allele_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Allele1&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">pval_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;p.value&quot;</span><span class="p">,</span>
<span class="p">)</span>
</code></pre></div>
<h3 id="harmonizing-data">Harmonizing data</h3>
<div class="highlight"><pre><span></span><code>harmonized_data &lt;- harmonise_data(clumped_exp,out_dat,action=1)
</code></pre></div>
<h3 id="perform-mr-analysis">Perform MR analysis</h3>
<div class="highlight"><pre><span></span><code><span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">mr</span><span class="p">(</span><span class="n">harmonized_data</span><span class="p">)</span>

<span class="n">id.exposure</span><span class="w"> </span><span class="n">id.outcome</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">method</span><span class="w">  </span><span class="n">nsnp</span><span class="w">    </span><span class="n">b</span><span class="w">   </span><span class="n">se</span><span class="w">  </span><span class="n">pval</span>
<span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">MR</span><span class="w"> </span><span class="n">Egger</span><span class="w">    </span><span class="m">28</span><span class="w">  </span><span class="m">1.3337580</span><span class="w">   </span><span class="m">0.69485260</span><span class="w">  </span><span class="m">6.596064e-02</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">Weighted</span><span class="w"> </span><span class="n">median</span><span class="w"> </span><span class="m">28</span><span class="w">  </span><span class="m">0.6298980</span><span class="w">   </span><span class="m">0.09401352</span><span class="w">  </span><span class="m">2.083081e-11</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">Inverse</span><span class="w"> </span><span class="n">variance</span><span class="w"> </span><span class="n">weighted</span><span class="w">   </span><span class="m">28</span><span class="w">  </span><span class="m">0.5598956</span><span class="w">   </span><span class="m">0.23225806</span><span class="w">  </span><span class="m">1.592361e-02</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">Simple</span><span class="w"> </span><span class="n">mode</span><span class="w"> </span><span class="m">28</span><span class="w">  </span><span class="m">0.6097842</span><span class="w">   </span><span class="m">0.15180476</span><span class="w">  </span><span class="m">4.232158e-04</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">Weighted</span><span class="w"> </span><span class="n">mode</span><span class="w">   </span><span class="m">28</span><span class="w">  </span><span class="m">0.5946778</span><span class="w">   </span><span class="m">0.12820220</span><span class="w">  </span><span class="m">8.044488e-05</span>
</code></pre></div>
<h2 id="sensitivity-analysis">Sensitivity analysis</h2>
<h3 id="heterogeneity">Heterogeneity</h3>
<div class="highlight"><pre><span></span><code><span class="nf">mr_heterogeneity</span><span class="p">(</span><span class="n">harmonized_data</span><span class="p">)</span>

<span class="n">id.exposure</span><span class="w"> </span><span class="n">id.outcome</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">method</span><span class="w">  </span><span class="n">Q</span><span class="w">   </span><span class="n">Q_df</span><span class="w">    </span><span class="n">Q_pval</span>
<span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">MR</span><span class="w"> </span><span class="n">Egger</span><span class="w">    </span><span class="m">670.7022</span><span class="w">    </span><span class="m">26</span><span class="w">  </span><span class="m">1.000684e-124</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">Inverse</span><span class="w"> </span><span class="n">variance</span><span class="w"> </span><span class="n">weighted</span><span class="w">   </span><span class="m">706.6579</span><span class="w">    </span><span class="m">27</span><span class="w">  </span><span class="m">1.534239e-131</span>
</code></pre></div>
<h3 id="horizontal-pleiotropy">Horizontal Pleiotropy</h3>
<p>Intercept in MR-Egger</p>
<div class="highlight"><pre><span></span><code><span class="nf">mr_pleiotropy_test</span><span class="p">(</span><span class="n">harmonized_data</span><span class="p">)</span>

<span class="n">id.exposure</span><span class="w"> </span><span class="n">id.outcome</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="n">egger_intercept</span><span class="w"> </span><span class="n">se</span><span class="w">  </span><span class="n">pval</span>
<span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">   </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span>
<span class="m">9</span><span class="n">J8pv4</span><span class="w">  </span><span class="n">IyUv6b</span><span class="w">  </span><span class="n">outcome</span><span class="w"> </span><span class="n">exposure</span><span class="w">    </span><span class="m">-0.03603697</span><span class="w"> </span><span class="m">0.0305241</span><span class="w">   </span><span class="m">0.2484472</span>
</code></pre></div>
<h3 id="single-snp-mr-and-leave-one-out-mr">Single SNP MR and leave-one-out MR</h3>
<p>Single SNP MR</p>
<div class="highlight"><pre><span></span><code>res_single &lt;- mr_singlesnp(harmonized_data)
res_single

exposure    outcome id.exposure id.outcome  samplesize  SNP b   se  p
&lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
1   exposure    outcome 9J8pv4  IyUv6b  NA  rs10198356  0.6323140   0.2082837   2.398742e-03
2   exposure    outcome 9J8pv4  IyUv6b  NA  rs10209994  0.9477808   0.3225814   3.302164e-03
3   exposure    outcome 9J8pv4  IyUv6b  NA  rs10824329  0.6281765   0.3246214   5.297739e-02
4   exposure    outcome 9J8pv4  IyUv6b  NA  rs10938397  1.2376316   0.2775854   8.251150e-06
5   exposure    outcome 9J8pv4  IyUv6b  NA  rs11066132  0.6024303   0.2232401   6.963693e-03
6   exposure    outcome 9J8pv4  IyUv6b  NA  rs12522139  0.2905201   0.2890240   3.148119e-01
7   exposure    outcome 9J8pv4  IyUv6b  NA  rs12591730  0.8930490   0.3076687   3.700413e-03
8   exposure    outcome 9J8pv4  IyUv6b  NA  rs13013021  1.4867889   0.2207777   1.646925e-11
9   exposure    outcome 9J8pv4  IyUv6b  NA  rs1955337   0.5442640   0.2994146   6.910079e-02
10  exposure    outcome 9J8pv4  IyUv6b  NA  rs2076308   1.1176226   0.2657969   2.613132e-05
11  exposure    outcome 9J8pv4  IyUv6b  NA  rs2278557   0.6238587   0.2968184   3.556906e-02
12  exposure    outcome 9J8pv4  IyUv6b  NA  rs2304608   1.5054682   0.2968905   3.961740e-07
13  exposure    outcome 9J8pv4  IyUv6b  NA  rs2531995   1.3972908   0.3130157   8.045689e-06
14  exposure    outcome 9J8pv4  IyUv6b  NA  rs261967    1.5303384   0.2921192   1.616714e-07
15  exposure    outcome 9J8pv4  IyUv6b  NA  rs35332469  -0.2307314  0.3479219   5.072217e-01
16  exposure    outcome 9J8pv4  IyUv6b  NA  rs35560038  -1.5730870  0.2018968   6.619637e-15
17  exposure    outcome 9J8pv4  IyUv6b  NA  rs3755804   0.5314915   0.2325073   2.225933e-02
18  exposure    outcome 9J8pv4  IyUv6b  NA  rs4470425   0.6948046   0.3079944   2.407689e-02
19  exposure    outcome 9J8pv4  IyUv6b  NA  rs476828    1.1739083   0.1568550   7.207355e-14
20  exposure    outcome 9J8pv4  IyUv6b  NA  rs4883723   0.5479721   0.2855004   5.494141e-02
21  exposure    outcome 9J8pv4  IyUv6b  NA  rs509325    0.5491040   0.1598196   5.908641e-04
22  exposure    outcome 9J8pv4  IyUv6b  NA  rs55872725  1.3501891   0.1259791   8.419325e-27
23  exposure    outcome 9J8pv4  IyUv6b  NA  rs6089309   0.5657525   0.3347009   9.096620e-02
24  exposure    outcome 9J8pv4  IyUv6b  NA  rs6265  0.6457693   0.1901871   6.851804e-04
25  exposure    outcome 9J8pv4  IyUv6b  NA  rs6736712   0.5606962   0.3448784   1.039966e-01
26  exposure    outcome 9J8pv4  IyUv6b  NA  rs7560832   0.6032080   0.2904972   3.785077e-02
27  exposure    outcome 9J8pv4  IyUv6b  NA  rs825486    -0.6152759  0.3500334   7.878772e-02
28  exposure    outcome 9J8pv4  IyUv6b  NA  rs9348441   -4.9786332  0.2572782   1.992909e-83
29  exposure    outcome 9J8pv4  IyUv6b  NA  All - Inverse variance weighted 0.5598956   0.2322581   1.592361e-02
30  exposure    outcome 9J8pv4  IyUv6b  NA  All - MR Egger  1.3337580   0.6948526   6.596064e-02
</code></pre></div>
<p>leave-one-out MR</p>
<div class="highlight"><pre><span></span><code>res_loo &lt;- mr_leaveoneout(harmonized_data)
res_loo

exposure    outcome id.exposure id.outcome  samplesize  SNP b   se  p
&lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
1   exposure    outcome 9J8pv4  IyUv6b  NA  rs10198356  0.5562834   0.2424917   2.178871e-02
2   exposure    outcome 9J8pv4  IyUv6b  NA  rs10209994  0.5520576   0.2388122   2.079526e-02
3   exposure    outcome 9J8pv4  IyUv6b  NA  rs10824329  0.5585335   0.2390239   1.945341e-02
4   exposure    outcome 9J8pv4  IyUv6b  NA  rs10938397  0.5412688   0.2388709   2.345460e-02
5   exposure    outcome 9J8pv4  IyUv6b  NA  rs11066132  0.5580606   0.2417275   2.096381e-02
6   exposure    outcome 9J8pv4  IyUv6b  NA  rs12522139  0.5667102   0.2395064   1.797373e-02
7   exposure    outcome 9J8pv4  IyUv6b  NA  rs12591730  0.5524802   0.2390990   2.085075e-02
8   exposure    outcome 9J8pv4  IyUv6b  NA  rs13013021  0.5189715   0.2386808   2.968017e-02
9   exposure    outcome 9J8pv4  IyUv6b  NA  rs1955337   0.5602635   0.2394505   1.929468e-02
10  exposure    outcome 9J8pv4  IyUv6b  NA  rs2076308   0.5431355   0.2394403   2.330758e-02
11  exposure    outcome 9J8pv4  IyUv6b  NA  rs2278557   0.5583634   0.2394924   1.972992e-02
12  exposure    outcome 9J8pv4  IyUv6b  NA  rs2304608   0.5372557   0.2377325   2.382639e-02
13  exposure    outcome 9J8pv4  IyUv6b  NA  rs2531995   0.5419016   0.2379712   2.277590e-02
14  exposure    outcome 9J8pv4  IyUv6b  NA  rs261967    0.5358761   0.2376686   2.415093e-02
15  exposure    outcome 9J8pv4  IyUv6b  NA  rs35332469  0.5735907   0.2378345   1.587739e-02
16  exposure    outcome 9J8pv4  IyUv6b  NA  rs35560038  0.6734906   0.2217804   2.391474e-03
17  exposure    outcome 9J8pv4  IyUv6b  NA  rs3755804   0.5610215   0.2413249   2.008503e-02
18  exposure    outcome 9J8pv4  IyUv6b  NA  rs4470425   0.5568993   0.2392632   1.993549e-02
19  exposure    outcome 9J8pv4  IyUv6b  NA  rs476828    0.5037555   0.2443224   3.922224e-02
20  exposure    outcome 9J8pv4  IyUv6b  NA  rs4883723   0.5602050   0.2397325   1.945000e-02
21  exposure    outcome 9J8pv4  IyUv6b  NA  rs509325    0.5608429   0.2468506   2.308693e-02
22  exposure    outcome 9J8pv4  IyUv6b  NA  rs55872725  0.4419446   0.2454771   7.180543e-02
23  exposure    outcome 9J8pv4  IyUv6b  NA  rs6089309   0.5597859   0.2388902   1.911519e-02
24  exposure    outcome 9J8pv4  IyUv6b  NA  rs6265  0.5547068   0.2436910   2.282978e-02
25  exposure    outcome 9J8pv4  IyUv6b  NA  rs6736712   0.5598815   0.2387602   1.902944e-02
26  exposure    outcome 9J8pv4  IyUv6b  NA  rs7560832   0.5588113   0.2396229   1.969836e-02
27  exposure    outcome 9J8pv4  IyUv6b  NA  rs825486    0.5800026   0.2367545   1.429330e-02
28  exposure    outcome 9J8pv4  IyUv6b  NA  rs9348441   0.7378967   0.1366838   6.717515e-08
29  exposure    outcome 9J8pv4  IyUv6b  NA  All 0.5598956   0.2322581   1.592361e-02
</code></pre></div>
<h2 id="visualization">Visualization</h2>
<h3 id="scatter-plot">Scatter plot</h3>
<div class="highlight"><pre><span></span><code>res &lt;- mr(harmonized_data)
p1 &lt;- mr_scatter_plot(res, harmonized_data)
p1[[1]]
</code></pre></div>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/214480227-396f816f-e1e6-49a1-9f3e-2e43a9d03abf.png" /></p>
<h3 id="single-snp">Single SNP</h3>
<div class="highlight"><pre><span></span><code>res_single &lt;- mr_singlesnp(harmonized_data)
p2 &lt;- mr_forest_plot(res_single)
p2[[1]]
</code></pre></div>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/214480253-6de266cf-2737-4d4f-b7fb-e889fea3ea4e.png" /></p>
<h3 id="leave-one-out">Leave-one-out</h3>
<div class="highlight"><pre><span></span><code>res_loo &lt;- mr_leaveoneout(harmonized_data)
p3 &lt;- mr_leaveoneout_plot(res_loo)
p3[[1]]
</code></pre></div>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/214480292-5bc318a3-1d74-4f09-8b07-b8a3cc62d2e0.png" /></p>
<h3 id="funnel-plot">Funnel plot</h3>
<div class="highlight"><pre><span></span><code>res_single &lt;- mr_singlesnp(harmonized_data)
p4 &lt;- mr_funnel_plot(res_single)
p4[[1]]
</code></pre></div>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/214480351-86288d44-52fb-416a-bdf4-2a4f4804e744.png" /></p>
<h2 id="mr-steiger-directionality-test">MR Steiger directionality test</h2>
<p>MR Steiger directionality test is a method to test the causal direction.</p>
<div class="highlight"><pre><span></span><code>harmonized_data$&quot;r.outcome&quot; &lt;- get_r_from_lor(
  harmonized_data$&quot;beta.outcome&quot;,
  harmonized_data$&quot;eaf.outcome&quot;,
  45383,
  132032,
  0.26,
  model = &quot;logit&quot;,
  correction = FALSE
)

out &lt;- directionality_test(harmonized_data)
out

id.exposure id.outcome  exposure    outcome snp_r2.exposure snp_r2.outcome  correct_causal_direction    steiger_pval
&lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;lgl&gt;   &lt;dbl&gt;
rvi6Om  ETcv15  BMI T2D 0.02125453  0.005496427 TRUE    NA
</code></pre></div>
<p>Reference: Hemani, G., Tilling, K., &amp; Davey Smith, G. (2017). Orienting the causal relationship between imprecisely measured traits using GWAS summary data. PLoS genetics, 13(11), e1007081.</p>
<h2 id="mr-base-web-app">MR-Base (web app)</h2>
<p><a href="https://app.mrbase.org/">MR-Base web app</a></p>
<h2 id="reference">Reference</h2>
<ul>
<li>Sanderson, E., Glymour, M. M., Holmes, M. V., Kang, H., Morrison, J., Munaf, M. R., ... &amp; Davey Smith, G. (2022). Mendelian randomization. Nature Reviews Methods Primers, 2(1), 1-21.</li>
<li>
<h2 id="hemani-g-zheng-j-elsworth-b-wade-k-h-haberland-v-baird-d-haycock-p-c-2018-the-mr-base-platform-supports-systematic-causal-inference-across-the-human-phenome-elife-7-e34408">Hemani, G., Zheng, J., Elsworth, B., Wade, K. H., Haberland, V., Baird, D., ... &amp; Haycock, P. C. (2018). The MR-Base platform supports systematic causal inference across the human phenome. elife, 7, e34408.</h2>
</li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      GWASTutorial is licensed under the MIT license
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.top", "navigation.tracking", "navigation.path"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.51198bba.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>