
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="HE Yunye">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-8.5.7">
    
    
      
        <title>LD socre regression by LDSC - GWASTutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.20d9efc8.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
          
          
          <meta name="theme-color" content="#2094f3">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#ld-score-regression" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GWASTutorial" class="md-header__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GWASTutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              LD socre regression by LDSC
            
          </span>
        </div>
      </div>
    </div>
    
    
    
    
      <div class="md-header__source">
        <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../02_Linux_basics/" class="md-tabs__link">
        Command Line
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../01_Dataset/" class="md-tabs__link">
        Pre-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../06_Association_tests/" class="md-tabs__link">
        GWAS
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../07_Annotation/" class="md-tabs__link md-tabs__link--active">
        Post-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../90_Recommended_Reading/" class="md-tabs__link">
        Others
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GWASTutorial" class="md-nav__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    GWASTutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_1">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Home" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Tutorial guide
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../99_About/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Command Line
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Command Line" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Command Line
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../02_Linux_basics/" class="md-nav__link">
        Linux Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../82_windows_linux_subsystem/" class="md-nav__link">
        Windows Linux Subsystem
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../83_git_and_github/" class="md-nav__link">
        Git and github
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../84_ssh/" class="md-nav__link">
        SSH
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../85_job_scheduler/" class="md-nav__link">
        Job scheduler
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../60_awk/" class="md-nav__link">
        Text editing - awk
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../61_sed/" class="md-nav__link">
        Text editing - sed
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../69_resources/" class="md-nav__link">
        Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Pre-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Pre-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../01_Dataset/" class="md-nav__link">
        Sample Dataset
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../03_Data_formats/" class="md-nav__link">
        Data Formats
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../04_Data_QC/" class="md-nav__link">
        Data QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../05_PCA/" class="md-nav__link">
        PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../plot_PCA/" class="md-nav__link">
        Plotting PCA
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../06_Association_tests/" class="md-nav__link">
        Association tests by PLINK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Visualization/" class="md-nav__link">
        Visualization by gwaslab
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../33_linear_mixed_model/" class="md-nav__link">
        Linear mixed model (LMM)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../32_whole_genome_regression/" class="md-nav__link">
        Whole genome regression by REGENIE
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Post-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Post-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Post-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../07_Annotation/" class="md-nav__link">
        Annotation by ANNOVAR & VEP
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../13_heritability/" class="md-nav__link">
        Heritability
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          LD socre regression by LDSC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        LD socre regression by LDSC
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav" aria-label="Introduction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ld-linkage-disequilibrium" class="md-nav__link">
    LD: Linkage disequilibrium
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-score" class="md-nav__link">
    LD score
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-score-regression_1" class="md-nav__link">
    LD score regression
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-ldsc" class="md-nav__link">
    Install LDSC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-preparation" class="md-nav__link">
    Data Preparation
  </a>
  
    <nav class="md-nav" aria-label="Data Preparation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-sample-summary-statistics" class="md-nav__link">
    Download sample summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-reference-files" class="md-nav__link">
    Download reference files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#munge-sumstats" class="md-nav__link">
    Munge sumstats
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ld-score-regression_2" class="md-nav__link">
    LD score regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cross-trait-ld-score-regression" class="md-nav__link">
    Cross-trait LD score regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#partitioned-ld-regression" class="md-nav__link">
    Partitioned LD regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#celltype-specificity-ld-regression" class="md-nav__link">
    Celltype specificity LD regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../09_Gene_based_analysis/" class="md-nav__link">
        Gene/Gene-set tests by MAGMA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../10_PRS/" class="md-nav__link">
        PRS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Others
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Others" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Others
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../90_Recommended_Reading/" class="md-nav__link">
        Recommended reading
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../80_anaconda/" class="md-nav__link">
        Anaconda
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../81_jupyter_notebook/" class="md-nav__link">
        Jupyter notebook
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav" aria-label="Introduction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ld-linkage-disequilibrium" class="md-nav__link">
    LD: Linkage disequilibrium
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-score" class="md-nav__link">
    LD score
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-score-regression_1" class="md-nav__link">
    LD score regression
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-ldsc" class="md-nav__link">
    Install LDSC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-preparation" class="md-nav__link">
    Data Preparation
  </a>
  
    <nav class="md-nav" aria-label="Data Preparation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-sample-summary-statistics" class="md-nav__link">
    Download sample summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-reference-files" class="md-nav__link">
    Download reference files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#munge-sumstats" class="md-nav__link">
    Munge sumstats
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ld-score-regression_2" class="md-nav__link">
    LD score regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cross-trait-ld-score-regression" class="md-nav__link">
    Cross-trait LD score regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#partitioned-ld-regression" class="md-nav__link">
    Partitioned LD regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#celltype-specificity-ld-regression" class="md-nav__link">
    Celltype specificity LD regression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="ld-score-regression">LD score regression</h1>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#install-ldsc">Install LDSC</a></li>
<li><a href="#data-preparation">Data Preparation</a></li>
<li><a href="#ld-score-regression">LD score regression</a></li>
<li><a href="#cross-trait-ld-score-regression">Cross-trait LD score regression</a></li>
<li><a href="#partitioned-ld-regression">Partitioned LD regression</a></li>
<li><a href="#celltype-specificity-ld-regression">Celltype specificity LD regression</a></li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>LDSC is one of the most commonly used command line tool to estimate hertability, genetic correlation and cell/tissue type specificity from GWAS summary statistics. </p>
<h3 id="ld-linkage-disequilibrium">LD: Linkage disequilibrium</h3>
<p>Linkage disequilibrium (LD) :  non-random association of alleles at different loci in a given population. (<a href="https://en.wikipedia.org/wiki/Linkage_disequilibrium">Wiki</a>)</p>
<h3 id="ld-score">LD score</h3>
<p>LD score <span class="arithmatex">\(l_j\)</span> for a SNP <span class="arithmatex">\(j\)</span> is defined as the sum of <span class="arithmatex">\(r^2\)</span> for the SNP and other SNPs in a region. </p>
<div class="arithmatex">\[
l_j= \Sigma_k{r^2_{j,k}}
\]</div>
<h3 id="ld-score-regression_1">LD score regression</h3>
<p>Key idea: A variant will have higher test statistics if it is in LD with causal variant, and the elevation is proportional to the LD ( <span class="arithmatex">\(r^2\)</span> ) with the causal variant.  </p>
<div class="arithmatex">\[
E[\chi^2|l_j] = {{Nh^2l_j}\over{M}} + Na + 1
\]</div>
<ul>
<li><span class="arithmatex">\(N\)</span>: sample size.</li>
<li><span class="arithmatex">\(M\)</span> : number of SNPs.</li>
<li><span class="arithmatex">\(h^2\)</span> : observed-scale heritability</li>
<li><span class="arithmatex">\(a\)</span> : the effect of confounding factors, including crytic relatedness and populatiuon stratification.</li>
</ul>
<p>For more details of LD score regression, please refer to :
- Bulik-Sullivan, Brendan K., et al. "LD Score regression distinguishes confounding from polygenicity in genome-wide association studies." Nature genetics 47.3 (2015): 291-295.</p>
<h2 id="install-ldsc">Install LDSC</h2>
<p>LDSC can be downloaded from github (GPL-3.0 license):
<a href="https://github.com/bulik/ldsc">https://github.com/bulik/ldsc</a></p>
<p>For ldsc, we need anaconda to create virtual environment. 
If you haven't installed Anaconda, please check <a href="https://cloufield.github.io/GWASTutorial/80_anaconda/">how to install anaconda</a>.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># change to your directory for tools</span>
<span class="nb">cd</span> ~/tools

<span class="c1"># clone the ldsc github repository</span>
git clone https://github.com/bulik/ldsc.git

<span class="c1"># create a virtual environment for ldsc (python2)</span>
<span class="nb">cd</span> ldsc
conda env create --file environment.yml  

<span class="c1"># activate ldsc environment</span>
conda activate ldsc
</code></pre></div>
<h2 id="data-preparation">Data Preparation</h2>
<p>In this tutoial, we will use sample summary statistics for HDLC and LDLC from Jenger. 
- Kanai, Masahiro, et al. "Genetic analysis of quantitative traits in the Japanese population links cell types to complex human diseases." Nature genetics 50.3 (2018): 390-400.</p>
<p>The Miami plot for the two traits:</p>
<p><img width="682" alt="image" src="https://user-images.githubusercontent.com/40289485/209749071-171c150a-19aa-41f0-b6e6-2ef5fa87370d.png"></p>
<h3 id="download-sample-summary-statistics">Download sample summary statistics</h3>
<div class="highlight"><pre><span></span><code><span class="c1"># HDL-c and LDL-c in Biobank Japan</span>
wget -O BBJ_LDLC.txt.gz http://jenger.riken.jp/61analysisresult_qtl_download/
wget -O BBJ_HDLC.txt.gz http://jenger.riken.jp/47analysisresult_qtl_download/
</code></pre></div>
<h3 id="download-reference-files">Download reference files</h3>
<p><div class="highlight"><pre><span></span><code><span class="c1"># change to your ldsc directory</span>
<span class="nb">cd</span> ~/tools/ldsc
mkdir resource
<span class="nb">cd</span> ./resource

<span class="c1"># snplist</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/w_hm3.snplist.bz2

<span class="c1"># EAS ld score files</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/eas_ldscores.tar.bz2

<span class="c1"># EAS weight</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/1000G_Phase3_EAS_weights_hm3_no_MHC.tgz

<span class="c1"># EAS frequency</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/1000G_Phase3_EAS_plinkfiles.tgz

<span class="c1"># EAS baseline model</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/1000G_Phase3_EAS_baseline_v1.2_ldscores.tgz

<span class="c1"># Cell type ld score files</span>
wget https://storage.googleapis.com/broad-alkesgroup-public/LDSCORE/LDSC_SEG_ldscores/Cahoy_EAS_1000Gv3_ldscores.tar.gz
</code></pre></div>
You can then decompress the files and organize them.</p>
<h2 id="munge-sumstats">Munge sumstats</h2>
<p>Before the analysis, we need to format and clean the raw sumstats.</p>
<div class="highlight"><pre><span></span><code><span class="nv">snplist</span><span class="o">=</span>~/tools/ldsc/resource/w_hm3.snplist
munge_sumstats.py <span class="se">\</span>
    --sumstats BBJ_HDLC.txt.gz <span class="se">\</span>
    --merge-alleles <span class="nv">$snplist</span> <span class="se">\</span>
    --a1 ALT <span class="se">\</span>
    --a2 REF <span class="se">\</span>
    --out BBJ_HDLC
munge_sumstats.py <span class="se">\</span>
  --sumstats BBJ_LDLC.txt.gz <span class="se">\</span>
    --a1 ALT <span class="se">\</span>
  --a2 REF <span class="se">\</span>
  --merge-alleles <span class="nv">$snplist</span> <span class="se">\</span>
  --out BBJ_LDLC
</code></pre></div>
<p>After munging, you will get two munged and formatted files:</p>
<p><div class="highlight"><pre><span></span><code>BBJ_HDLC.sumstats.gz
BBJ_LDLC.sumstats.gz
</code></pre></div>
And these are the files we will use to run LD score regression.</p>
<h2 id="ld-score-regression_2">LD score regression</h2>
<p>Univariate LD score regression is utilized to estimate heritbility and confuding factors (cryptic relateness and population stratification) of a certain trait. </p>
<ul>
<li>Reference : Bulik-Sullivan, Brendan K., et al. "LD Score regression distinguishes confounding from polygenicity in genome-wide association studies." Nature genetics 47.3 (2015): 291-295.</li>
</ul>
<p>Using the munged sumstats, we can run:</p>
<div class="highlight"><pre><span></span><code>ldsc.py <span class="se">\</span>
  --h2 BBJ_HDLC.sumstats.gz <span class="se">\</span>
  --ref-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --w-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --out BBJ_HDLC

ldsc.py <span class="se">\</span>
  --h2 BBJ_LDLC.sumstats.gz <span class="se">\</span>
  --ref-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --w-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --out BBJ_LDLC
</code></pre></div>
<p>Lest's check the results for HDLC:</p>
<div class="highlight"><pre><span></span><code>cat BBJ_HDLC.log
*********************************************************************
* LD Score Regression <span class="o">(</span>LDSC<span class="o">)</span>
* Version <span class="m">1</span>.0.1
* <span class="o">(</span>C<span class="o">)</span> <span class="m">2014</span>-2019 Brendan Bulik-Sullivan and Hilary Finucane
* Broad Institute of MIT and Harvard / MIT Department of Mathematics
* GNU General Public License v3
*********************************************************************
Call: 
./ldsc.py <span class="se">\</span>
--h2 BBJ_HDLC.sumstats.gz <span class="se">\</span>
--ref-ld-chr /home/he/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
--out BBJ_HDLC <span class="se">\</span>
--w-ld-chr /home/he/tools/ldsc/resource/eas_ldscores/ 

Beginning analysis at Sat Dec <span class="m">24</span> <span class="m">20</span>:40:34 <span class="m">2022</span>
Reading summary statistics from BBJ_HDLC.sumstats.gz ...
Read summary statistics <span class="k">for</span> <span class="m">1020377</span> SNPs.
Reading reference panel LD Score from /home/he/tools/ldsc/resource/eas_ldscores/<span class="o">[</span><span class="m">1</span>-22<span class="o">]</span> ... <span class="o">(</span>ldscore_fromlist<span class="o">)</span>
Read reference panel LD Scores <span class="k">for</span> <span class="m">1208050</span> SNPs.
Removing partitioned LD Scores with zero variance.
Reading regression weight LD Score from /home/he/tools/ldsc/resource/eas_ldscores/<span class="o">[</span><span class="m">1</span>-22<span class="o">]</span> ... <span class="o">(</span>ldscore_fromlist<span class="o">)</span>
Read regression weight LD Scores <span class="k">for</span> <span class="m">1208050</span> SNPs.
After merging with reference panel LD, <span class="m">1012040</span> SNPs remain.
After merging with regression SNP LD, <span class="m">1012040</span> SNPs remain.
Using two-step estimator with cutoff at <span class="m">30</span>.
Total Observed scale h2: <span class="m">0</span>.1583 <span class="o">(</span><span class="m">0</span>.0281<span class="o">)</span>
Lambda GC: <span class="m">1</span>.1523
Mean Chi^2: <span class="m">1</span>.2843
Intercept: <span class="m">1</span>.0563 <span class="o">(</span><span class="m">0</span>.0114<span class="o">)</span>
Ratio: <span class="m">0</span>.1981 <span class="o">(</span><span class="m">0</span>.0402<span class="o">)</span>
Analysis finished at Sat Dec <span class="m">24</span> <span class="m">20</span>:40:41 <span class="m">2022</span>
Total <span class="nb">time</span> elapsed: <span class="m">6</span>.57s
</code></pre></div>
<p>We can see that from the log:</p>
<ul>
<li>Observed scale h2 = 0.1583</li>
<li>lambda GC = 1.1523 </li>
<li>intercept = 1.0563</li>
<li>Ratio = 0.1981</li>
</ul>
<p>According to LDSC documents, Ratio measures the proportion of the inflation in the mean chi^2 that the LD Score regression intercept ascribes to causes other than polygenic heritability. The value of ratio should be close to zero, though in practice values of 10-20% are not uncommon.</p>
<div class="arithmatex">\[
Ratio = {{intercept-1}\over{mean(\chi^2)-1}}
\]</div>
<h2 id="cross-trait-ld-score-regression">Cross-trait LD score regression</h2>
<p>Cross-trait LD score regression is employed to estimate the genetic correlation between a pair of traits.</p>
<p>Key idea: replace <code>\chi^2</code> in univariate LD score regression and the relationship (SNPs with high LD ) still holds.</p>
<div class="arithmatex">\[
E[z_{1j}z_{2j}] = {{\sqrt{N_1N_2}\rho_g}\over{M}}l_j + {{\rho N_s}\over{\sqrt{N_1N_2}}}
\]</div>
<ul>
<li><span class="arithmatex">\(z_ij\)</span> : z score of trait i for SNP j</li>
<li><span class="arithmatex">\(N_i\)</span> : sample size of trait i</li>
<li><span class="arithmatex">\(\rho\)</span> : phenotypic correlation</li>
<li><span class="arithmatex">\(\rho_g\)</span> : genetic covariance </li>
<li><span class="arithmatex">\(l_j\)</span> : LD score</li>
<li><span class="arithmatex">\(M\)</span> : number of SNPs</li>
</ul>
<p>Then we can get the genetic correlation by :</p>
<div class="arithmatex">\[
r_g = {{\rho_g}\over{\sqrt{h_1^2h_2^2}}}
\]</div>
<ul>
<li>Reference: Bulik-Sullivan, Brendan, et al. "An atlas of genetic correlations across human diseases and traits." Nature genetics 47.11 (2015): 1236-1241.</li>
</ul>
<p><div class="highlight"><pre><span></span><code>ldsc.py <span class="se">\</span>
  --rg BBJ_HDLC.sumstats.gz,BBJ_LDLC.sumstats.gz <span class="se">\</span>
  --ref-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --w-ld-chr ~/tools/ldsc/resource/eas_ldscores/ <span class="se">\</span>
  --out BBJ_HDLC_LDLC
</code></pre></div>
Let's check the results:</p>
<div class="highlight"><pre><span></span><code>*********************************************************************
* LD Score Regression (LDSC)
* Version 1.0.1
* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
* Broad Institute of MIT and Harvard / MIT Department of Mathematics
* GNU General Public License v3
*********************************************************************
Call: 
./ldsc.py \
--ref-ld-chr /home/he/tools/ldsc/resource/eas_ldscores/ \
--out BBJ_HDLC_LDLC \
--rg BBJ_HDLC.sumstats.gz,BBJ_LDLC.sumstats.gz \
--w-ld-chr /home/he/tools/ldsc/resource/eas_ldscores/ 

Beginning analysis at Thu Dec 29 21:02:37 2022
Reading summary statistics from BBJ_HDLC.sumstats.gz ...
Read summary statistics for 1020377 SNPs.
Reading reference panel LD Score from /home/he/tools/ldsc/resource/eas_ldscores/[1-22] ... (ldscore_fromlist)
Read reference panel LD Scores for 1208050 SNPs.
Removing partitioned LD Scores with zero variance.
Reading regression weight LD Score from /home/he/tools/ldsc/resource/eas_ldscores/[1-22] ... (ldscore_fromlist)
Read regression weight LD Scores for 1208050 SNPs.
After merging with reference panel LD, 1012040 SNPs remain.
After merging with regression SNP LD, 1012040 SNPs remain.
Computing rg for phenotype 2/2
Reading summary statistics from BBJ_LDLC.sumstats.gz ...
Read summary statistics for 1217311 SNPs.
After merging with summary statistics, 1012040 SNPs remain.
1012040 SNPs with valid alleles.

Heritability of phenotype 1
---------------------------
Total Observed scale h2: 0.1054 (0.0383)
Lambda GC: 1.1523
Mean Chi^2: 1.2843
Intercept: 1.1234 (0.0607)
Ratio: 0.4342 (0.2134)

Heritability of phenotype 2/2
-----------------------------
Total Observed scale h2: 0.0543 (0.0211)
Lambda GC: 1.0833
Mean Chi^2: 1.1465
Intercept: 1.0583 (0.0335)
Ratio: 0.398 (0.2286)

Genetic Covariance
------------------
Total Observed scale gencov: 0.0121 (0.0106)
Mean z1*z2: -0.001
Intercept: -0.0198 (0.0121)

Genetic Correlation
-------------------
Genetic Correlation: 0.1601 (0.1821)
Z-score: 0.8794
P: 0.3792


Summary of Genetic Correlation Results
p1                    p2      rg      se       z       p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se
BBJ_HDLC.sumstats.gz  BBJ_LDLC.sumstats.gz  0.1601  0.1821  0.8794  0.3792  0.0543     0.0211  1.0583     0.0335   -0.0198       0.0121

Analysis finished at Thu Dec 29 21:02:47 2022
Total time elapsed: 10.39s
</code></pre></div>
<h2 id="partitioned-ld-regression">Partitioned LD regression</h2>
<p>Partitioned LD regression is utilized to evaluate the contribution of each functional group to the total SNP heriatbility.</p>
<div class="arithmatex">\[
E[\chi^2] = N \sum\limits_C \tau_C l(j,C) + Na + 1
\]</div>
<ul>
<li><span class="arithmatex">\(C\)</span> : functional categories</li>
<li><span class="arithmatex">\(l(j,C)\)</span> : LD score for SNP j with respect to C. </li>
<li>
<p><span class="arithmatex">\(\tau_C\)</span> : per-SNP contribution of category C to heritability</p>
</li>
<li>
<p>Reference: Finucane, Hilary K., et al. "Partitioning heritability by functional annotation using genome-wide association summary statistics." Nature genetics 47.11 (2015): 1228-1235.</p>
</li>
</ul>
<div class="highlight"><pre><span></span><code>ldsc.py <span class="se">\</span>
  --h2 BBJ_HDLC.sumstats.gz <span class="se">\</span>
  --overlap-annot <span class="se">\</span>
  --ref-ld-chr ~/tools/ldsc/resource/1000G_Phase3_EAS_baseline_v1_2_ldscores/baseline. <span class="se">\</span>
  --frqfile-chr ~/tools/ldsc/resource/1000G_Phase3_EAS_plinkfiles/1000G.EAS.QC. <span class="se">\</span>
  --w-ld-chr ~/tools/ldsc/resource/1000G_Phase3_EAS_weights_hm3_no_MHC/weights.EAS.hm3_noMHC. <span class="se">\</span>
  --out BBJ_HDLC_baseline
</code></pre></div>
<h2 id="celltype-specificity-ld-regression">Celltype specificity LD regression</h2>
<p>LDSC-SEG :  LD score regression applied to specifically expressed genes</p>
<p>An extension of Partitioned LD regression. Categories are defined by tissue or cell-type specific genes.</p>
<ul>
<li>Reference: Finucane, Hilary K., et al. "Heritability enrichment of specifically expressed genes identifies disease-relevant tissues and cell types." Nature genetics 50.4 (2018): 621-629.</li>
</ul>
<div class="highlight"><pre><span></span><code>ldsc.py <span class="se">\</span>
  --h2-cts BBJ_HDLC.sumstats.gz <span class="se">\</span>
  --ref-ld-chr-cts ~/tools/ldsc/resource/Cahoy_EAS_1000Gv3_ldscores/Cahoy.EAS.ldcts <span class="se">\</span>
  --ref-ld-chr ~/tools/ldsc/resource/1000G_Phase3_EAS_baseline_v1_2_ldscores/baseline. <span class="se">\</span>
  --w-ld-chr ~/tools/ldsc/resource/1000G_Phase3_EAS_weights_hm3_no_MHC/weights.EAS.hm3_noMHC. <span class="se">\</span>
  --out BBJ_HDLC_baseline_cts
</code></pre></div>
<h2 id="reference">Reference</h2>
<ul>
<li>Bulik-Sullivan, Brendan K., et al. "LD Score regression distinguishes confounding from polygenicity in genome-wide association studies." Nature genetics 47.3 (2015): 291-295.</li>
<li>Bulik-Sullivan, Brendan, et al. "An atlas of genetic correlations across human diseases and traits." Nature genetics 47.11 (2015): 1236-1241.</li>
<li>Finucane, Hilary K., et al. "Partitioning heritability by functional annotation using genome-wide association summary statistics." Nature genetics 47.11 (2015): 1228-1235.</li>
<li>Finucane, Hilary K., et al. "Heritability enrichment of specifically expressed genes identifies disease-relevant tissues and cell types." Nature genetics 50.4 (2018): 621-629.</li>
<li>Kanai, Masahiro, et al. "Genetic analysis of quantitative traits in the Japanese population links cell types to complex human diseases." Nature genetics 50.3 (2018): 390-400.</li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../13_heritability/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Heritability" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Heritability
            </div>
          </div>
        </a>
      
      
        
        <a href="../09_Gene_based_analysis/" class="md-footer__link md-footer__link--next" aria-label="Next: Gene/Gene-set tests by MAGMA" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Gene/Gene-set tests by MAGMA
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      GWASTutorial is licensed under the MIT license
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.8492ddcf.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>