
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="HE Yunye">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-8.5.10">
    
    
      
        <title>Data QC - GWASTutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.975780f9.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.2505c338.min.css">
        
          
          
          <meta name="theme-color" content="#2094f3">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#plink-basics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GWASTutorial" class="md-header__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GWASTutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Data QC
            
          </span>
        </div>
      </div>
    </div>
    
    
    
    
      <div class="md-header__source">
        <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../02_Linux_basics/" class="md-tabs__link">
        Command Line
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../01_Dataset/" class="md-tabs__link md-tabs__link--active">
        Pre-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../06_Association_tests/" class="md-tabs__link">
        GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../07_Annotation/" class="md-tabs__link">
        Post-GWAS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../90_Recommended_Reading/" class="md-tabs__link">
        Others
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GWASTutorial" class="md-nav__button md-logo" aria-label="GWASTutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    GWASTutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Cloufield/GWASTutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_1">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Home" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Tutorial guide
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../99_About/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Command Line
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Command Line" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Command Line
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../02_Linux_basics/" class="md-nav__link">
        Linux Basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../82_windows_linux_subsystem/" class="md-nav__link">
        Windows Linux Subsystem
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../83_git_and_github/" class="md-nav__link">
        Git and github
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../84_ssh/" class="md-nav__link">
        SSH
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../85_job_scheduler/" class="md-nav__link">
        Job scheduler
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../60_awk/" class="md-nav__link">
        Text editing - awk
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../61_sed/" class="md-nav__link">
        Text editing - sed
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../80_anaconda/" class="md-nav__link">
        Anaconda
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../81_jupyter_notebook/" class="md-nav__link">
        Jupyter notebook
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../69_resources/" class="md-nav__link">
        Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Pre-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Pre-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../01_Dataset/" class="md-nav__link">
        Sample Dataset
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../03_Data_formats/" class="md-nav__link">
        Data Formats
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Data QC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Data QC
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparation" class="md-nav__link">
    Preparation
  </a>
  
    <nav class="md-nav" aria-label="Preparation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#plink-192-installation" class="md-nav__link">
    PLINK 1.9&amp;2 installation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-plink19-and-plink2-and-then-unzip" class="md-nav__link">
    Download PLINK1.9 and PLINK2 and then unzip
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-symbolic-links" class="md-nav__link">
    Create symbolic links
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink19" class="md-nav__link">
    PLINK1.9
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink2" class="md-nav__link">
    PLINK2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-genotype-data" class="md-nav__link">
    Download genotype data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plink-tutorial" class="md-nav__link">
    PLINK tutorial
  </a>
  
    <nav class="md-nav" aria-label="PLINK tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#missing-rate-call-rate" class="md-nav__link">
    Missing rate (call rate)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink-syntax" class="md-nav__link">
    PLink syntax:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#allele-frequency" class="md-nav__link">
    Allele Frequency
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inbreeding-f-coefficient" class="md-nav__link">
    Inbreeding F coefficient
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hardy-weinberg-equilibrium-exact-test" class="md-nav__link">
    Hardy-Weinberg equilibrium exact test
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#applying-filters" class="md-nav__link">
    Applying filters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pruning" class="md-nav__link">
    Pruning
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-snp-filtering-extractexcludekeepremove" class="md-nav__link">
    Sample &amp; SNP filtering (extract/exclude/keep/remove)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ibd-pi_hat" class="md-nav__link">
    IBD / PI_HAT
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-calculation" class="md-nav__link">
    LD calculation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-management-make-bedrecode" class="md-nav__link">
    Data management (make-bed/recode)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercise" class="md-nav__link">
    Exercise
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-resources" class="md-nav__link">
    Additional resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../05_PCA/" class="md-nav__link">
        PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../plot_PCA/" class="md-nav__link">
        Plotting PCA
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../06_Association_tests/" class="md-nav__link">
        Association tests by PLINK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Visualization/" class="md-nav__link">
        Visualization by gwaslab
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../33_linear_mixed_model/" class="md-nav__link">
        Linear mixed model (LMM)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../32_whole_genome_regression/" class="md-nav__link">
        Whole genome regression by REGENIE
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Post-GWAS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Post-GWAS" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Post-GWAS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../07_Annotation/" class="md-nav__link">
        Annotation by ANNOVAR & VEP
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../13_heritability/" class="md-nav__link">
        Heritability
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../08_LDSC/" class="md-nav__link">
        LD socre regression by LDSC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../09_Gene_based_analysis/" class="md-nav__link">
        Gene/Gene-set tests by MAGMA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../10_PRS/" class="md-nav__link">
        PRS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Others
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Others" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Others
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../90_Recommended_Reading/" class="md-nav__link">
        Recommended reading
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../95_Assignment/" class="md-nav__link">
        Assignment - 1KG PCA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../96_Assignment2/" class="md-nav__link">
        Assignment - Presentation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparation" class="md-nav__link">
    Preparation
  </a>
  
    <nav class="md-nav" aria-label="Preparation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#plink-192-installation" class="md-nav__link">
    PLINK 1.9&amp;2 installation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-plink19-and-plink2-and-then-unzip" class="md-nav__link">
    Download PLINK1.9 and PLINK2 and then unzip
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-symbolic-links" class="md-nav__link">
    Create symbolic links
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink19" class="md-nav__link">
    PLINK1.9
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink2" class="md-nav__link">
    PLINK2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#download-genotype-data" class="md-nav__link">
    Download genotype data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plink-tutorial" class="md-nav__link">
    PLINK tutorial
  </a>
  
    <nav class="md-nav" aria-label="PLINK tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#missing-rate-call-rate" class="md-nav__link">
    Missing rate (call rate)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plink-syntax" class="md-nav__link">
    PLink syntax:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#allele-frequency" class="md-nav__link">
    Allele Frequency
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inbreeding-f-coefficient" class="md-nav__link">
    Inbreeding F coefficient
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hardy-weinberg-equilibrium-exact-test" class="md-nav__link">
    Hardy-Weinberg equilibrium exact test
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#applying-filters" class="md-nav__link">
    Applying filters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pruning" class="md-nav__link">
    Pruning
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-snp-filtering-extractexcludekeepremove" class="md-nav__link">
    Sample &amp; SNP filtering (extract/exclude/keep/remove)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ibd-pi_hat" class="md-nav__link">
    IBD / PI_HAT
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ld-calculation" class="md-nav__link">
    LD calculation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-management-make-bedrecode" class="md-nav__link">
    Data management (make-bed/recode)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercise" class="md-nav__link">
    Exercise
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-resources" class="md-nav__link">
    Additional resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="plink-basics">Plink basics</h1>
<p>In this module, we will learn the basics of genotype data QC using PLINK, which is one of the most commonly used software in complex trait genomics.</p>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><a href="#preparation">Preparation</a><ul>
<li><a href="#plink-192-installation">PLINK 1.9 &amp; 2 installation</a></li>
<li><a href="#download-genotype-data">Download genotype data</a></li>
</ul>
</li>
<li><a href="#plink-tutorial">PLINK tutorial</a><ul>
<li><a href="#missing-rate-call-rate">Calculate missing rate and call rate</a></li>
<li><a href="#allele-frequency">Calculate allele frequency</a></li>
<li><a href="#inbreeding-f-coefficient">Calculate inbreeding F coefficient </a></li>
<li><a href="#hardy-weinberg-equilibrium-exact-test">Hardy-Weinberg equilibrium exact test</a></li>
<li><a href="#applying-filters">Applying filters</a></li>
<li><a href="#pruning">LD-Pruning</a></li>
<li><a href="#sample--snp-filtering-extractexcludekeepremove">Sample &amp; SNP filtering (extract/exclude/keep/remove)</a></li>
<li><a href="#ld-calculation">LD calculation</a></li>
<li><a href="#ibd--pi_hat">Estimate IBD / PI_HAT</a></li>
<li><a href="#data-management-make-bedrecode">Data management (make-bed/recode)</a></li>
</ul>
</li>
<li><a href="#exercise">Exercise</a></li>
<li><a href="#additional-resources">Additional resources</a></li>
<li><a href="#reference">Reference</a></li>
</ul>
<h2 id="preparation">Preparation</h2>
<h3 id="plink-192-installation">PLINK 1.9&amp;2 installation</h3>
<p>To get prepared for genotype QC, we will need to make directories, download softwares and add the sofwares to your environment path.</p>
<p>First, we will simply create some directories to keep the tools we need to use.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~
mkdir tools
<span class="nb">cd</span> tools
mkdir bin
mkdir plink
mkdir plink2
</code></pre></div>
<p>You can download each tool into its corresponding directories. </p>
<p>The <code>bin</code> directory here is for keeping all the symbolic links to the executable files of each tool. </p>
<p>In this way, it is much easier to mangnage and organize the paths and tools.</p>
<p><img width="937" alt="image" src="https://user-images.githubusercontent.com/40289485/160745597-c6f4204a-d786-4af1-9041-f1531cbbe584.png"></p>
<h3 id="download-plink19-and-plink2-and-then-unzip">Download PLINK1.9 and PLINK2 and then unzip</h3>
<p>Next, go to the plink webpage to download the software. We will need both PLINK1.9 and PLINK2.</p>
<p>Download PLINK1.9 and PLINK2 from the following webpage to the corresponding directories:
https://www.cog-genomics.org/plink/
https://www.cog-genomics.org/plink/2.0/</p>
<p>Note: If you are using mac or windows, then please download the mac or windows version. </p>
<p>In this tutorial we will use a linux system. </p>
<p>Find the suitable version, right click and copy the link address.</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> plink2
wget https://s3.amazonaws.com/plink2-assets/plink2_linux_x86_64_20211011.zip
unzip plink2_linux_x86_64_20211011.zip
</code></pre></div>
Then do the same for PLINK1.9</p>
<h3 id="create-symbolic-links">Create symbolic links</h3>
<p>After donwloading and unzipping, we will create symbolic links for the plink binany files, and then move the link to <code>~/tools/bin/</code>.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~
ln -s ~/tools/plink2/plink2 ~/tools/bin/plink2
ln -s ~/tools/plink/plink ~/tools/bin/plink
</code></pre></div></p>
<p>Then add <code>~/tools/bin/</code> to the environment path.</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">export</span> <span class="nv">PATH</span><span class="o">=</span><span class="nv">$PATH</span>:~/tools/bin/
</code></pre></div>
And then run 
<div class="highlight"><pre><span></span><code>echo &quot;export PATH=$PATH:~/tools/bin/&quot; &gt;&gt; ~/.bashrc
</code></pre></div>
This will add a new line at the end of <code>.bashrc</code>, which will be run every time you open a new bash shell.</p>
<p>All done. Let's test if we installed PLINK successfully or not.</p>
<div class="highlight"><pre><span></span><code>plink -h
plink2 -h
</code></pre></div>
<h3 id="plink19">PLINK1.9</h3>
<div class="highlight"><pre><span></span><code>plink -h
PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3

  plink &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]
  plink --help [flag name(s)...]

Commands include --make-bed, --recode, --flip-scan, --merge-list,
--write-snplist, --list-duplicate-vars, --freqx, --missing, --test-mishap,
--hardy, --mendel, --ibc, --impute-sex, --indep-pairphase, --r2, --show-tags,
--blocks, --distance, --genome, --homozyg, --make-rel, --make-grm-gz,
--rel-cutoff, --cluster, --pca, --neighbour, --ibs-test, --regress-distance,
--model, --bd, --gxe, --logistic, --dosage, --lasso, --test-missing,
--make-perm-pheno, --tdt, --qfam, --annotate, --clump, --gene-report,
--meta-analysis, --epistasis, --fast-epistasis, and --score.

&quot;plink --help | more&quot; describes all functions (warning: long).
</code></pre></div>
<h3 id="plink2">PLINK2</h3>
<p><div class="highlight"><pre><span></span><code>plink2 -h
PLINK v2.00a3LM 64-bit Intel (1 Jul 2021)      www.cog-genomics.org/plink/2.0/
(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3

  plink2 &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]
  plink2 --help [flag name(s)...]

Commands include --rm-dup list, --make-bpgen, --export, --freq, --geno-counts,
--sample-counts, --missing, --hardy, --het, --fst, --indep-pairwise, --ld,
--sample-diff, --make-king, --king-cutoff, --pmerge, --pgen-diff,
--write-samples, --write-snplist, --make-grm-list, --pca, --glm, --adjust-file,
--score, --variant-score, --genotyping-rate, --pgen-info, --validate, and
--zst-decompress.

&quot;plink2 --help | more&quot; describes all functions.
</code></pre></div>
Well done. We have installed plink1.9 and plink2.</p>
<h3 id="download-genotype-data">Download genotype data</h3>
<p>Next we need to download the sample genotype data.</p>
<p>Run the download.sh in 01_Dataset.</p>
<div class="highlight"><pre><span></span><code>cd ../01_Dataset
./download_sampledata.sh
</code></pre></div>
<p>And you will get the following three files:</p>
<div class="highlight"><pre><span></span><code>-rw-r-----   1 he  staff   135M Dec 23 11:45 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bed
-rw-r-----   1 he  staff    36M Dec 23 11:46 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim
-rw-r-----   1 he  staff   9.4K Dec 23 11:46 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam
</code></pre></div>
<p>Check the bim file:</p>
<div class="highlight"><pre><span></span><code>head 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim
<span class="m">1</span>   <span class="m">1</span>:13273:G:C <span class="m">0</span>   <span class="m">13273</span>   C   G
<span class="m">1</span>   <span class="m">1</span>:14599:T:A <span class="m">0</span>   <span class="m">14599</span>   A   T
<span class="m">1</span>   <span class="m">1</span>:14604:A:G <span class="m">0</span>   <span class="m">14604</span>   G   A
<span class="m">1</span>   <span class="m">1</span>:14930:A:G <span class="m">0</span>   <span class="m">14930</span>   G   A
<span class="m">1</span>   <span class="m">1</span>:69897:T:C <span class="m">0</span>   <span class="m">69897</span>   C   T
<span class="m">1</span>   <span class="m">1</span>:86331:A:G <span class="m">0</span>   <span class="m">86331</span>   G   A
<span class="m">1</span>   <span class="m">1</span>:91581:G:A <span class="m">0</span>   <span class="m">91581</span>   A   G
<span class="m">1</span>   <span class="m">1</span>:122872:T:G    <span class="m">0</span>   <span class="m">122872</span>  G   T
<span class="m">1</span>   <span class="m">1</span>:135163:C:T    <span class="m">0</span>   <span class="m">135163</span>  T   C
<span class="m">1</span>   <span class="m">1</span>:233473:C:G    <span class="m">0</span>   <span class="m">233473</span>  G   C
</code></pre></div>
<p>Check the fam file:
<div class="highlight"><pre><span></span><code>head 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam
<span class="m">0</span> HG00403 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00404 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00406 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00407 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00409 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00410 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00419 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00421 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00422 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
<span class="m">0</span> HG00428 <span class="m">0</span> <span class="m">0</span> <span class="m">0</span> -9
</code></pre></div></p>
<h2 id="plink-tutorial">PLINK tutorial</h2>
<p>Detailed description can be found on plink's website: <a href="https://www.cog-genomics.org/plink/1.9/">PLINK1.9</a> and <a href="https://www.cog-genomics.org/plink/2.0/">PLINK2</a>.</p>
<p>The functions we will learn in this tutorial:
1. Calculating missing rate (call rate)
2. Calculating inbreeding F coefficient
3. Conducting Hardy-Weinberg equilibrium exact test
4. Calculating allele Frequency
5. Applying filters
6. Conducting LD-Pruning
7. Conducting sample &amp; snp filtering (extract/exclude/keep/remove)
8. Estimating IBD / PI_HAT
9. Calculating LD
10. Data management (make-bed/recode)</p>
<p>All sample codes and results for this module are available in <code>./04_data_QC</code></p>
<p>First, we can calculate some basic statistics of our simulated data:</p>
<h3 id="missing-rate-call-rate">Missing rate (call rate)</h3>
<p>The first thing we want to know is the missing rate of our data. Usually we need to check the missing rate of samples and snps to decide a threshold to exclude low-quality samples and snps. (https://www.cog-genomics.org/plink/1.9/basic_stats#missing)</p>
<ul>
<li><strong>Sample missing rate</strong>: the proportion of missing values for an individual across all SNPSs.</li>
<li><strong>SNP missing rate</strong>: the proportion of missing values for a SNP across all samples.</li>
</ul>
<p>The input is PLINK bed/bim/fam file. Usually they have the same prefix, and we just need to pass the prefix to <code>--bfile</code> option.</p>
<h3 id="plink-syntax">PLink syntax:</h3>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/161413684-a128b87f-fb79-4b13-a7b5-acfa997c4421.png" /></p>
<p>To calculate the missing rate, we need the flag <code>--missing</code>, which tells PLINK to calculate the missing rate in the dataset specified by <code>--bfile</code>. </p>
<p>Sample code to calculated missing rate:</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ../04_Data_QC
<span class="nv">genotypeFile</span><span class="o">=</span><span class="s2">&quot;../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot;</span> <span class="c1">#!!! Please add your own path here.  &quot;1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot; is the prefix of PLINK bed file. </span>

plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --missing <span class="se">\</span>
    --out plink_results
</code></pre></div>
Remeber to set the value for <code>${genotypeFile}</code>.</p>
<p>This code will generate two files <code>plink_results.imiss</code> and <code>plink_results.lmiss</code>, which contain the missing rate information for samples and snps respectively.</p>
<p>Take a look at the imiss file. The last column shows the missing rate for samples. Since we used part of the 1000 Genome Project data this time, there are no missing samples or snps. So the missing rate is zero.</p>
<div class="highlight"><pre><span></span><code>head plink_results.imiss
 FID       IID MISS_PHENO   N_MISS   N_GENO   F_MISS
   <span class="m">0</span>   HG00403          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00404          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00406          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00407          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00409          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00410          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00419          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00421          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
   <span class="m">0</span>   HG00422          Y        <span class="m">0</span>  <span class="m">1122299</span>        <span class="m">0</span>
</code></pre></div>
<p>For the meaning of headers, please refer to <a href="https://www.cog-genomics.org/plink/1.9/formats">PLINK documents</a>.</p>
<h3 id="allele-frequency">Allele Frequency</h3>
<p>One of the most important statistics of SNPs is their frequencies in a certain population. A number of downstream analysis is based on investigating differences in allele frequencies.</p>
<p>Usually, variants can be categorized into 3 groups based on their Minor Allel Frequency (MAF):</p>
<ol>
<li><strong>Common variants</strong> : MAF&gt;=0.05</li>
<li><strong>Low-frequency variants</strong> : 0.01&lt;=MAF&lt;0.05</li>
<li><strong>Rare variants</strong> : MAF&lt;0.01</li>
</ol>
<p>For different downstream analyses, we might use different set of variants. For example, for PCA, we might use only common variants. For gene-based tests, we might use only rare variants.</p>
<p>Using PLINK1.9 we can easily calculate the MAF of variants in the input data.</p>
<div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --freq <span class="se">\</span>
    --out plink_results
</code></pre></div>
<p>Next we use plink2 to run the same options to check the difference between the results.
<div class="highlight"><pre><span></span><code>plink2 <span class="se">\</span>
        --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
        --freq <span class="se">\</span>
        --out plink_results
</code></pre></div></p>
<div class="highlight"><pre><span></span><code><span class="c1"># results from plink1.9</span>
head plink_results.frq
 CHR              SNP   A1   A2          MAF  NCHROBS
   <span class="m">1</span>      <span class="m">1</span>:13273:G:C    C    G       <span class="m">0</span>.0625     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:14599:T:A    A    T      <span class="m">0</span>.08929     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:14604:A:G    G    A      <span class="m">0</span>.08929     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:14930:A:G    G    A       <span class="m">0</span>.4137     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:69897:T:C    T    C        <span class="m">0</span>.124     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:86331:A:G    G    A       <span class="m">0</span>.0873     <span class="m">1008</span>
   <span class="m">1</span>      <span class="m">1</span>:91581:G:A    A    G        <span class="m">0</span>.498     <span class="m">1008</span>
   <span class="m">1</span>     <span class="m">1</span>:122872:T:G    G    T       <span class="m">0</span>.2589     <span class="m">1008</span>
   <span class="m">1</span>     <span class="m">1</span>:135163:C:T    T    C      <span class="m">0</span>.09226     <span class="m">1008</span>
</code></pre></div>
<p><div class="highlight"><pre><span></span><code><span class="c1"># results from plink2</span>
head plink_results.afreq
<span class="c1">#CHROM  ID  REF ALT ALT_FREQS   OBS_CT</span>
<span class="m">1</span>   <span class="m">1</span>:13273:G:C G   C   <span class="m">0</span>.0625  <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:14599:T:A T   A   <span class="m">0</span>.0892857   <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:14604:A:G A   G   <span class="m">0</span>.0892857   <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:14930:A:G A   G   <span class="m">0</span>.41369 <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:69897:T:C T   C   <span class="m">0</span>.875992    <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:86331:A:G A   G   <span class="m">0</span>.0873016   <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:91581:G:A G   A   <span class="m">0</span>.498016    <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:122872:T:G    T   G   <span class="m">0</span>.258929    <span class="m">1008</span>
<span class="m">1</span>   <span class="m">1</span>:135163:C:T    C   T   <span class="m">0</span>.0922619   <span class="m">1008</span>
</code></pre></div>
We need pay attention to the concepts here.</p>
<p>In PLINK1.9, the concept here is minor (A1) and major(A2) allele, while in PLINK2 it is reference(REF) and altervative(ALT) allele.</p>
<ul>
<li><strong>Major / Minor</strong>: Major allele and minor allele are defined as the allele with highest and lower(or second highest for multiallelic variants) allele in a given population, respectively. So major allele and minor allele for a SNP might be different in two independent populations. The range for MAF(minor allele frequencies) is [0,0.5].</li>
<li><strong>Ref / Alt</strong>: Reference(REF) and altervative(ALT) alleles are simply determined by the allele on a reference genome. If we use the same reference genome, the reference(REF) and altervative(ALT) allele will be the same acorss populations. Reference allele could be major or minor in different populations. The range for alternative allele frequency is [0,1], since it could be major or minor allele in a given population.</li>
</ul>
<h3 id="inbreeding-f-coefficient">Inbreeding F coefficient</h3>
<p>Next we can check the heterozygosity F of samples (https://www.cog-genomics.org/plink/1.9/basic_stats#ibc) : </p>
<p><code>-het</code> option will computes observed and expected autosomal homozygous genotype counts for each sample. Usually, we need to exclude individuals with high or low heterozygosity coefficient, which suggests that the sample might be contaminated. </p>
<div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --het <span class="se">\</span>
    --out plink_results
</code></pre></div>
<p>Check the output:</p>
<p><div class="highlight"><pre><span></span><code>head plink_results.het
 FID       IID       O<span class="o">(</span>HOM<span class="o">)</span>       E<span class="o">(</span>HOM<span class="o">)</span>        N<span class="o">(</span>NM<span class="o">)</span>            F
   <span class="m">0</span>   HG00403       <span class="m">747270</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>    -0.004114
   <span class="m">0</span>   HG00404       <span class="m">748955</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>    <span class="m">0</span>.0003974
   <span class="m">0</span>   HG00406       <span class="m">750093</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>     <span class="m">0</span>.003444
   <span class="m">0</span>   HG00407       <span class="m">746566</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>    -0.005999
   <span class="m">0</span>   HG00409       <span class="m">751694</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>     <span class="m">0</span>.007731
   <span class="m">0</span>   HG00410       <span class="m">745078</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>    -0.009983
   <span class="m">0</span>   HG00419       <span class="m">747996</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>     -0.00217
   <span class="m">0</span>   HG00421       <span class="m">757199</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>      <span class="m">0</span>.02247
   <span class="m">0</span>   HG00422       <span class="m">752725</span>    <span class="m">7</span>.488e+05      <span class="m">1122299</span>      <span class="m">0</span>.01049
</code></pre></div>
A commonly used method is to exclude samples with heterzygosity F deviating more than 3 standard dividation(SD) from the mean F.</p>
<h3 id="hardy-weinberg-equilibrium-exact-test">Hardy-Weinberg equilibrium exact test</h3>
<p>For SNP QC, besides checking the missing rate, we also need to check if the SNP is in hardy-weinberg equilibrium:</p>
<p><code>--hardy</code> will perform Hardy-Weinberg equilibrium exact test for each variant. Variants with low P value usually suggest genotyping errors, or indicate evolutionary selection for these variants.</p>
<p>The following command can calculate the Hardy-Weinberg equilibrium exact test statistics for all SNPs. (https://www.cog-genomics.org/plink/1.9/basic_stats#hardy)
<div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --hardy <span class="se">\</span>
    --out plink_results
</code></pre></div>
<div class="highlight"><pre><span></span><code>head plink_results.hwe
 CHR              SNP     TEST   A1   A2                 GENO   O<span class="o">(</span>HET<span class="o">)</span>   E<span class="o">(</span>HET<span class="o">)</span>            P 
   <span class="m">1</span>      <span class="m">1</span>:13273:G:C  ALL<span class="o">(</span>NP<span class="o">)</span>    C    G             <span class="m">1</span>/61/442    <span class="m">0</span>.121   <span class="m">0</span>.1172       <span class="m">0</span>.7113
   <span class="m">1</span>      <span class="m">1</span>:14599:T:A  ALL<span class="o">(</span>NP<span class="o">)</span>    A    T             <span class="m">1</span>/88/415   <span class="m">0</span>.1746   <span class="m">0</span>.1626       <span class="m">0</span>.1625
   <span class="m">1</span>      <span class="m">1</span>:14604:A:G  ALL<span class="o">(</span>NP<span class="o">)</span>    G    A             <span class="m">1</span>/88/415   <span class="m">0</span>.1746   <span class="m">0</span>.1626       <span class="m">0</span>.1625
   <span class="m">1</span>      <span class="m">1</span>:14930:A:G  ALL<span class="o">(</span>NP<span class="o">)</span>    G    A             <span class="m">4</span>/409/91   <span class="m">0</span>.8115   <span class="m">0</span>.4851    <span class="m">1</span>.679e-61
   <span class="m">1</span>      <span class="m">1</span>:69897:T:C  ALL<span class="o">(</span>NP<span class="o">)</span>    T    C            <span class="m">7</span>/111/386   <span class="m">0</span>.2202   <span class="m">0</span>.2173            <span class="m">1</span>
   <span class="m">1</span>      <span class="m">1</span>:86331:A:G  ALL<span class="o">(</span>NP<span class="o">)</span>    G    A             <span class="m">0</span>/88/416   <span class="m">0</span>.1746   <span class="m">0</span>.1594      <span class="m">0</span>.02387
   <span class="m">1</span>      <span class="m">1</span>:91581:G:A  ALL<span class="o">(</span>NP<span class="o">)</span>    A    G          <span class="m">137</span>/228/139   <span class="m">0</span>.4524      <span class="m">0</span>.5      <span class="m">0</span>.03271
   <span class="m">1</span>     <span class="m">1</span>:122872:T:G  ALL<span class="o">(</span>NP<span class="o">)</span>    G    T            <span class="m">1</span>/259/244   <span class="m">0</span>.5139   <span class="m">0</span>.3838     <span class="m">8</span>.04e-19
   <span class="m">1</span>     <span class="m">1</span>:135163:C:T  ALL<span class="o">(</span>NP<span class="o">)</span>    T    C             <span class="m">1</span>/91/412   <span class="m">0</span>.1806   <span class="m">0</span>.1675       <span class="m">0</span>.1066
</code></pre></div></p>
<h3 id="applying-filters">Applying filters</h3>
<p>Previously we just calculate the basic statistics using PLINK. But when performing certain analyses, we just want to exclude the bad quallity samples or SNPs instead of calculating the statistics for all samples and SNPs.</p>
<p>In this case we can apply the following filters for example:</p>
<ul>
<li><code>--maf 0.01</code> : exlcude snps with maf&lt;0.01</li>
<li><code>--geno 0.01</code> :filters out all variants with missing rates exceeding 0.0</li>
<li><code>--mind 0.02</code> :filters out all samples with missing rates exceeding 0.02</li>
<li><code>--hwe 5e-6</code> : filters out all variants which have Hardy-Weinberg equilibrium exact test p-value below the provided threshold. NOTE: With case/control data, cases and missing phenotypes are normally ignored. (see https://www.cog-genomics.org/plink/1.9/filter#hwe)</li>
</ul>
<h3 id="pruning">Pruning</h3>
<p>There are ofter strong Linkage disequilibrium(LD) among SNPs, for some analysis we don't need all SNPs and we need to remove the redundant SNPs to avoid bias in genetic estimations. For example, for relatedness estimation, we will use only LD-pruned SNP set. </p>
<p>We can use <code>--indep-pairwise 50 5 0.2</code> to filter out those are in strong LD and keep only the independent SNPS.
Please check https://www.cog-genomics.org/plink/1.9/ld#indep for the meaning of each parameter.
Combined with the filters we just introduced, we can run:</p>
<p><div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --maf <span class="m">0</span>.01 <span class="se">\</span>
    --geno <span class="m">0</span>.01 <span class="se">\</span>
    --mind <span class="m">0</span>.02 <span class="se">\</span>
    --hwe 5e-6 <span class="se">\</span>
    --indep-pairwise <span class="m">50</span> <span class="m">5</span> <span class="m">0</span>.2 <span class="se">\</span>
    --out plink_results
</code></pre></div>
This command generates two outputs :  <code>plink_results.prune.in</code> and <code>plink_results.prune.out</code>
<code>plink_results.prune.in</code> is the independent set of SNPs we will use in the following analysis.
Let's take a look at this file. Basically it just contains one SNP id per line.</p>
<div class="highlight"><pre><span></span><code>head plink_results.prune.in
<span class="m">1</span>:13273:G:C
<span class="m">1</span>:14599:T:A
<span class="m">1</span>:69897:T:C
<span class="m">1</span>:86331:A:G
<span class="m">1</span>:91581:G:A
<span class="m">1</span>:135163:C:T
<span class="m">1</span>:233473:C:G
<span class="m">1</span>:532929:T:C
<span class="m">1</span>:559480:C:T
<span class="m">1</span>:565433:C:T
</code></pre></div>
<h3 id="sample-snp-filtering-extractexcludekeepremove">Sample &amp; SNP filtering (extract/exclude/keep/remove)</h3>
<p>Some time we will use only a subset of samples or snps included the original dataset. 
In this case, we can use <code>--extract</code> or <code>--exclude</code> to select or exclude SNPs from analysis, <code>--keep</code> or <code>--remove</code> to select or exclude samples.</p>
<p>For  <code>--keep</code> or <code>--remove</code> , the input is the filename of a sample FID and IID file.
For <code>--extract</code> or <code>--exclude</code> , the input is the filename of a snplist file.
<div class="highlight"><pre><span></span><code>head plink_results.prune.in
<span class="m">1</span>:13273:G:C
<span class="m">1</span>:14599:T:A
<span class="m">1</span>:69897:T:C
<span class="m">1</span>:86331:A:G
<span class="m">1</span>:91581:G:A
<span class="m">1</span>:135163:C:T
<span class="m">1</span>:233473:C:G
<span class="m">1</span>:532929:T:C
<span class="m">1</span>:559480:C:T
<span class="m">1</span>:565433:C:T
</code></pre></div></p>
<h3 id="ibd-pi_hat">IBD / PI_HAT</h3>
<p><code>--genome</code> will estimate IBS/IBD. Usually for this analysis, we need to prune our data first since the strong LD will cause bias in the results.
(This step is extremelly computationally intensive)</p>
<p>Combined with the <code>--extract</code>, we can run:</p>
<div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
        --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --extract plink_results.prune.in <span class="se">\</span>
        --genome <span class="se">\</span>
        --out plink_results
</code></pre></div>
<p>PI_HAT is the IBD estimation. Please check https://www.cog-genomics.org/plink/1.9/ibd for more details.
<div class="highlight"><pre><span></span><code>head plink_results.genome
 FID1     IID1 FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00404 OT     <span class="m">0</span>  <span class="m">0</span>.9800  <span class="m">0</span>.0086  <span class="m">0</span>.0114  <span class="m">0</span>.0157  -1  <span class="m">0</span>.749375  <span class="m">0</span>.5531  <span class="m">2</span>.0089
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00406 OT     <span class="m">0</span>  <span class="m">0</span>.9751  <span class="m">0</span>.0231  <span class="m">0</span>.0018  <span class="m">0</span>.0133  -1  <span class="m">0</span>.748336  <span class="m">0</span>.6309  <span class="m">2</span>.0225
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00407 OT     <span class="m">0</span>  <span class="m">0</span>.9801  <span class="m">0</span>.0153  <span class="m">0</span>.0046  <span class="m">0</span>.0122  -1  <span class="m">0</span>.748293  <span class="m">0</span>.5045  <span class="m">2</span>.0007
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00409 OT     <span class="m">0</span>  <span class="m">0</span>.9807  <span class="m">0</span>.0193  <span class="m">0</span>.0000  <span class="m">0</span>.0097  -1  <span class="m">0</span>.747215  <span class="m">0</span>.6014  <span class="m">2</span>.0173
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00410 OT     <span class="m">0</span>  <span class="m">0</span>.9744  <span class="m">0</span>.0256  <span class="m">0</span>.0000  <span class="m">0</span>.0128  -1  <span class="m">0</span>.748058  <span class="m">0</span>.8621  <span class="m">2</span>.0745
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00419 OT     <span class="m">0</span>  <span class="m">0</span>.9650  <span class="m">0</span>.0350  <span class="m">0</span>.0000  <span class="m">0</span>.0175  -1  <span class="m">0</span>.747146  <span class="m">0</span>.8080  <span class="m">2</span>.0595
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00421 OT     <span class="m">0</span>  <span class="m">0</span>.9842  <span class="m">0</span>.0158  <span class="m">0</span>.0000  <span class="m">0</span>.0079  -1  <span class="m">0</span>.746522  <span class="m">0</span>.4557  <span class="m">1</span>.9926
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00422 OT     <span class="m">0</span>  <span class="m">0</span>.9832  <span class="m">0</span>.0111  <span class="m">0</span>.0056  <span class="m">0</span>.0112  -1  <span class="m">0</span>.748156  <span class="m">0</span>.4208  <span class="m">1</span>.9867
   <span class="m">0</span>  HG00403   <span class="m">0</span>  HG00428 OT     <span class="m">0</span>  <span class="m">0</span>.9929  <span class="m">0</span>.0000  <span class="m">0</span>.0071  <span class="m">0</span>.0071  -1  <span class="m">0</span>.746851  <span class="m">0</span>.3115  <span class="m">1</span>.9674
</code></pre></div></p>
<h3 id="ld-calculation">LD calculation</h3>
<p>We can also use our data to calculate LD between a pair of SNPs.</p>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/161413586-d4f6e21e-f0c7-4c54-a703-bb060ec6913d.png" /></p>
<p><code>--chr</code> option allows us to include snps on a specific chromosome.
To calculate LD r2 for SNPs on chr22 , we can run:
<div class="highlight"><pre><span></span><code>plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
        --chr <span class="m">22</span> <span class="se">\</span>
        --r2 <span class="se">\</span>
        --out plink_results
</code></pre></div>
<div class="highlight"><pre><span></span><code>head plink_results.ld
 CHR_A         BP_A             SNP_A  CHR_B         BP_B             SNP_B           R2 
    <span class="m">22</span>     <span class="m">16053659</span>   <span class="m">22</span>:16053659:A:C     <span class="m">22</span>     <span class="m">16067500</span>   <span class="m">22</span>:16067500:T:C      <span class="m">0</span>.64577 
    <span class="m">22</span>     <span class="m">16053862</span>   <span class="m">22</span>:16053862:C:T     <span class="m">22</span>     <span class="m">16054454</span>   <span class="m">22</span>:16054454:C:T     <span class="m">0</span>.987505 
    <span class="m">22</span>     <span class="m">16053862</span>   <span class="m">22</span>:16053862:C:T     <span class="m">22</span>     <span class="m">16055122</span>   <span class="m">22</span>:16055122:G:T     <span class="m">0</span>.601708 
    <span class="m">22</span>     <span class="m">16053863</span>   <span class="m">22</span>:16053863:G:A     <span class="m">22</span>     <span class="m">16055070</span>   <span class="m">22</span>:16055070:G:A     <span class="m">0</span>.931043 
    <span class="m">22</span>     <span class="m">16054454</span>   <span class="m">22</span>:16054454:C:T     <span class="m">22</span>     <span class="m">16055122</span>   <span class="m">22</span>:16055122:G:T     <span class="m">0</span>.594216 
    <span class="m">22</span>     <span class="m">16058767</span>   <span class="m">22</span>:16058767:A:G     <span class="m">22</span>     <span class="m">16067462</span>   <span class="m">22</span>:16067462:C:T     <span class="m">0</span>.266916 
    <span class="m">22</span>     <span class="m">16067462</span>   <span class="m">22</span>:16067462:C:T     <span class="m">22</span>     <span class="m">16069771</span>   <span class="m">22</span>:16069771:G:A     <span class="m">0</span>.382323 
    <span class="m">22</span>     <span class="m">16069771</span>   <span class="m">22</span>:16069771:G:A     <span class="m">22</span>     <span class="m">16123813</span>   <span class="m">22</span>:16123813:G:A     <span class="m">0</span>.291935 
    <span class="m">22</span>     <span class="m">16143946</span>   <span class="m">22</span>:16143946:A:G     <span class="m">22</span>     <span class="m">16155259</span>   <span class="m">22</span>:16155259:A:G     <span class="m">0</span>.559165 
</code></pre></div></p>
<h3 id="data-management-make-bedrecode">Data management (make-bed/recode)</h3>
<p>By far the input data we use is in binary form, but sometimes we may want the text version.
<img alt="image" src="https://user-images.githubusercontent.com/40289485/161413659-a489b508-63c7-4166-9f5c-25a1a125109a.png" /></p>
<p>To convert the format we can run:</p>
<div class="highlight"><pre><span></span><code><span class="c1">#extract the 1000 samples with the pruned SNPs, and make a bed file.</span>
plink <span class="se">\</span>
    --bfile <span class="si">${</span><span class="nv">genotypeFile</span><span class="si">}</span> <span class="se">\</span>
    --extract plink_results.prune.in <span class="se">\</span>
    --make-bed <span class="se">\</span>
    --out plink_1000_pruned

<span class="c1">#convert the bed/bim/fam to ped/map</span>
plink <span class="se">\</span>
        --bfile plink_1000_pruned <span class="se">\</span>
        --recode <span class="se">\</span>
        --out plink_1000_pruned
</code></pre></div>
<h2 id="exercise">Exercise</h2>
<ul>
<li>Follow this tutorial and type in the commands:</li>
<li>Calculate basic statistics for the simulated data.</li>
<li>
<p>Learn the meaning of each QC step.</p>
</li>
<li>
<p>Visualize the results of QC (using Python or R)</p>
</li>
<li>Draw the distribution of MAF.(histogram)</li>
<li>Draw the distribution of het.(histogram)</li>
<li>Try to briefly explain what you observe</li>
</ul>
<h2 id="additional-resources">Additional resources</h2>
<ul>
<li>Marees, A. T., de Kluiver, H., Stringer, S., Vorspan, F., Curis, E., MarieClaire, C., &amp; Derks, E. M. (2018). A tutorial on conducting genomewide association studies: Quality control and statistical analysis. International journal of methods in psychiatric research, 27(2), e1608.</li>
</ul>
<h2 id="reference">Reference</h2>
<ul>
<li>Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A., Bender, D., ... &amp; Sham, P. C. (2007). PLINK: a tool set for whole-genome association and population-based linkage analyses. The American journal of human genetics, 81(3), 559-575.</li>
<li>Chang, C. C., Chow, C. C., Tellier, L. C., Vattikuti, S., Purcell, S. M., &amp; Lee, J. J. (2015). Second-generation PLINK: rising to the challenge of larger and richer datasets. Gigascience, 4(1), s13742-015.</li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../03_Data_formats/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Data Formats" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Data Formats
            </div>
          </div>
        </a>
      
      
        
        <a href="../05_PCA/" class="md-footer__link md-footer__link--next" aria-label="Next: PCA" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              PCA
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      GWASTutorial is licensed under the MIT license
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.top"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.5a2dcb6a.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
    
  </body>
</html>