<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="HE Yunye">
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Data QC - GWASTutorial</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">GWASTutorial</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="../02_Linux_basics/" class="nav-link">Linux</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Pre-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../01_Dataset/" class="dropdown-item">Sample Dataset</a>
</li>
                                    
<li>
    <a href="../03_Data_formats/" class="dropdown-item">Data Formats</a>
</li>
                                    
<li>
    <a href="./" class="dropdown-item active">Data QC</a>
</li>
                                    
<li>
    <a href="../05_PCA/" class="dropdown-item">PCA</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../06_Association_tests/" class="dropdown-item">Association tests</a>
</li>
                                    
<li>
    <a href="../Visualization/" class="dropdown-item">Visualization</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Post-GWAS <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../07_Annotation/" class="dropdown-item">Annotation</a>
</li>
                                    
<li>
    <a href="../08_LDSC/" class="dropdown-item">LDSC</a>
</li>
                                    
<li>
    <a href="../09_Gene_based_analysis/" class="dropdown-item">Gene/Gene-set tests</a>
</li>
                                    
<li>
    <a href="../10_PRS/" class="dropdown-item">PRS</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Others <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../80_anaconda/" class="dropdown-item">Anaconda</a>
</li>
                                    
<li>
    <a href="../81_jupyter_notebook/" class="dropdown-item">Jupyter notebook</a>
</li>
                                    
<li>
    <a href="../82_windows_linux_subsystem/" class="dropdown-item">WLS</a>
</li>
                                    
<li>
    <a href="../83_git_and_github/" class="dropdown-item">Git and github</a>
</li>
                                    
<li>
    <a href="../60_awk/" class="dropdown-item">awk</a>
</li>
                                    
<li>
    <a href="../61_sed/" class="dropdown-item">sed</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                            <li class="nav-item">
                                <a rel="prev" href="../03_Data_formats/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../05_PCA/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/Cloufield/GWASTutorial" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#plink-basics" class="nav-link">Plink basics</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#table-of-contents" class="nav-link">Table of Contents</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#preparation" class="nav-link">Preparation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#plink-tutorial" class="nav-link">PLINK tutorial</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#exercise" class="nav-link">Exercise</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#additional-resources" class="nav-link">Additional resources</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#reference" class="nav-link">Reference</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="plink-basics">Plink basics</h1>
<p>In this module, we will learn the basics of genotype data QC using PLINK, which is one of the most commonly used software in complex trait genomics.</p>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><a href="#preparation">Preparation</a><ul>
<li><a href="#plink-192-installation">PLINK 1.9 &amp; 2 installation</a></li>
<li><a href="#download-genotype-data">Download genotype data</a></li>
</ul>
</li>
<li><a href="#plink-tutorial">PLINK tutorial</a><ul>
<li><a href="#missing-rate-call-rate">Calculate missing rate and call rate</a></li>
<li><a href="#allele-frequency">Calculate allele frequency</a></li>
<li><a href="#inbreeding-f-coefficient">Calculate inbreeding F coefficient </a></li>
<li><a href="#hardy-weinberg-equilibrium-exact-test">Hardy-Weinberg equilibrium exact test</a></li>
<li><a href="#applying-filters">Applying filters</a></li>
<li><a href="#pruning">LD-Pruning</a></li>
<li><a href="#sample--snp-filtering-extractexcludekeepremove">Sample &amp; SNP filtering (extract/exclude/keep/remove)</a></li>
<li><a href="#ld-calculation">LD calculation</a></li>
<li><a href="#ibd--pi_hat">Estimate IBD / PI_HAT</a></li>
<li><a href="#data-management-make-bedrecode">Data management (make-bed/recode)</a></li>
</ul>
</li>
<li><a href="#exercise">Exercise</a></li>
<li><a href="#additional-resources">Additional resources</a></li>
<li><a href="#reference">Reference</a></li>
</ul>
<h2 id="preparation">Preparation</h2>
<h3 id="plink-192-installation">PLINK 1.9&amp;2 installation</h3>
<p>To get prepared for genotype QC, we will need to make directories, download softwares and add the sofwares to your environment path.</p>
<p>First, we will simply create some directories to keep the tools we need to use.</p>
<pre><code class="language-bash">cd ~
mkdir tools
cd tools
mkdir bin
mkdir plink
mkdir plink2
</code></pre>
<p>You can download each tool into its corresponding directories. </p>
<p>The <code>bin</code> directory here is for keeping all the symbolic links to the executable files of each tool. </p>
<p>In this way, it is much easier to mangnage and organize the paths and tools.</p>
<p><img width="937" alt="image" src="https://user-images.githubusercontent.com/40289485/160745597-c6f4204a-d786-4af1-9041-f1531cbbe584.png"></p>
<h3 id="download-plink19-and-plink2-and-then-unzip">Download PLINK1.9 and PLINK2 and then unzip</h3>
<p>Next, go to the plink webpage to download the software. We will need both PLINK1.9 and PLINK2.</p>
<p>Download PLINK1.9 and PLINK2 from the following webpage to the corresponding directories:
https://www.cog-genomics.org/plink/
https://www.cog-genomics.org/plink/2.0/</p>
<p>Note: If you are using mac or windows, then please download the mac or windows version. </p>
<p>In this tutorial we will use a linux system. </p>
<p>Find the suitable version, right click and copy the link address.</p>
<pre><code class="language-bash">cd plink2
wget https://s3.amazonaws.com/plink2-assets/plink2_linux_x86_64_20211011.zip
unzip plink2_linux_x86_64_20211011.zip
</code></pre>
<p>Then do the same for PLINK1.9</p>
<h3 id="create-symbolic-links">Create symbolic links</h3>
<p>After donwloading and unzipping, we will create symbolic links for the plink binany files, and then move the link to <code>~/tools/bin/</code>.</p>
<pre><code class="language-bash">cd ~
ln -s ~/tools/plink2/plink2 ~/tools/bin/plink2
ln -s ~/tools/plink/plink ~/tools/bin/plink
</code></pre>
<p>Then add <code>~/tools/bin/</code> to the environment path.</p>
<pre><code class="language-bash">export PATH=$PATH:~/tools/bin/
</code></pre>
<p>And then run </p>
<pre><code>echo &quot;export PATH=$PATH:~/tools/bin/&quot; &gt;&gt; ~/.bashrc
</code></pre>
<p>This will add a new line at the end of <code>.bashrc</code>, which will be run every time you open a new bash shell.</p>
<p>All done. Let's test if we installed PLINK successfully or not.</p>
<pre><code class="language-bash">plink -h
plink2 -h
</code></pre>
<h3 id="plink19">PLINK1.9</h3>
<pre><code>plink -h
PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3

  plink &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]
  plink --help [flag name(s)...]

Commands include --make-bed, --recode, --flip-scan, --merge-list,
--write-snplist, --list-duplicate-vars, --freqx, --missing, --test-mishap,
--hardy, --mendel, --ibc, --impute-sex, --indep-pairphase, --r2, --show-tags,
--blocks, --distance, --genome, --homozyg, --make-rel, --make-grm-gz,
--rel-cutoff, --cluster, --pca, --neighbour, --ibs-test, --regress-distance,
--model, --bd, --gxe, --logistic, --dosage, --lasso, --test-missing,
--make-perm-pheno, --tdt, --qfam, --annotate, --clump, --gene-report,
--meta-analysis, --epistasis, --fast-epistasis, and --score.

&quot;plink --help | more&quot; describes all functions (warning: long).
</code></pre>
<h3 id="plink2">PLINK2</h3>
<pre><code>plink2 -h
PLINK v2.00a3LM 64-bit Intel (1 Jul 2021)      www.cog-genomics.org/plink/2.0/
(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3

  plink2 &lt;input flag(s)...&gt; [command flag(s)...] [other flag(s)...]
  plink2 --help [flag name(s)...]

Commands include --rm-dup list, --make-bpgen, --export, --freq, --geno-counts,
--sample-counts, --missing, --hardy, --het, --fst, --indep-pairwise, --ld,
--sample-diff, --make-king, --king-cutoff, --pmerge, --pgen-diff,
--write-samples, --write-snplist, --make-grm-list, --pca, --glm, --adjust-file,
--score, --variant-score, --genotyping-rate, --pgen-info, --validate, and
--zst-decompress.

&quot;plink2 --help | more&quot; describes all functions.

</code></pre>
<p>Well done. We have installed plink1.9 and plink2.</p>
<h3 id="download-genotype-data">Download genotype data</h3>
<p>Next we need to download the sample genotype data.</p>
<p>Run the download.sh in 01_Dataset.</p>
<pre><code>cd ../01_Dataset
./download_sampledata.sh
</code></pre>
<p>And you will get the following three files:</p>
<pre><code>-rw-r-----   1 he  staff   135M Dec 23 11:45 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bed
-rw-r-----   1 he  staff    36M Dec 23 11:46 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim
-rw-r-----   1 he  staff   9.4K Dec 23 11:46 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam
</code></pre>
<p>Check the bim file:</p>
<pre><code class="language-bash">head 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.bim
1   1:13273:G:C 0   13273   C   G
1   1:14599:T:A 0   14599   A   T
1   1:14604:A:G 0   14604   G   A
1   1:14930:A:G 0   14930   G   A
1   1:69897:T:C 0   69897   C   T
1   1:86331:A:G 0   86331   G   A
1   1:91581:G:A 0   91581   A   G
1   1:122872:T:G    0   122872  G   T
1   1:135163:C:T    0   135163  T   C
1   1:233473:C:G    0   233473  G   C
</code></pre>
<p>Check the fam file:</p>
<pre><code class="language-bash">head 1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020.fam
0 HG00403 0 0 0 -9
0 HG00404 0 0 0 -9
0 HG00406 0 0 0 -9
0 HG00407 0 0 0 -9
0 HG00409 0 0 0 -9
0 HG00410 0 0 0 -9
0 HG00419 0 0 0 -9
0 HG00421 0 0 0 -9
0 HG00422 0 0 0 -9
0 HG00428 0 0 0 -9
</code></pre>
<h2 id="plink-tutorial">PLINK tutorial</h2>
<p>Detailed description can be found on plink's website: <a href="https://www.cog-genomics.org/plink/1.9/">PLINK1.9</a> and <a href="https://www.cog-genomics.org/plink/2.0/">PLINK2</a>.</p>
<p>The functions we will learn in this tutorial:
1. Calculating missing rate (call rate)
2. Calculating inbreeding F coefficient
3. Conducting Hardy-Weinberg equilibrium exact test
4. Calculating allele Frequency
5. Applying filters
6. Conducting LD-Pruning
7. Conducting sample &amp; snp filtering (extract/exclude/keep/remove)
8. Estimating IBD / PI_HAT
9. Calculating LD
10. Data management (make-bed/recode)</p>
<p>All sample codes and results for this module are available in <code>./04_data_QC</code></p>
<p>First, we can calculate some basic statistics of our simulated data:</p>
<h3 id="missing-rate-call-rate">Missing rate (call rate)</h3>
<p>The first thing we want to know is the missing rate of our data. Usually we need to check the missing rate of samples and snps to decide a threshold to exclude low-quality samples and snps. (https://www.cog-genomics.org/plink/1.9/basic_stats#missing)</p>
<ul>
<li><strong>Sample missing rate</strong>: the proportion of missing values for an individual across all SNPSs.</li>
<li><strong>SNP missing rate</strong>: the proportion of missing values for a SNP across all samples.</li>
</ul>
<p>The input is PLINK bed/bim/fam file. Usually they have the same prefix, and we just need to pass the prefix to <code>--bfile</code> option.</p>
<h3 id="plink-syntax">PLink syntax:</h3>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/161413684-a128b87f-fb79-4b13-a7b5-acfa997c4421.png" /></p>
<p>To calculate the missing rate, we need the flag <code>--missing</code>, which tells PLINK to calculate the missing rate in the dataset specified by <code>--bfile</code>. </p>
<p>Sample code to calculated missing rate:</p>
<pre><code class="language-bash">cd ../04_Data_QC
genotypeFile=&quot;../01_Dataset/1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot; #!!! Please add your own path here.  &quot;1KG.EAS.auto.snp.norm.nodup.split.maf005.thinp020&quot; is the prefix of PLINK bed file. 

plink \
    --bfile ${genotypeFile} \
    --missing \
    --out plink_results
</code></pre>
<p>Remeber to set the value for <code>${genotypeFile}</code>.</p>
<p>This code will generate two files <code>plink_results.imiss</code> and <code>plink_results.lmiss</code>, which contain the missing rate information for samples and snps respectively.</p>
<p>Take a look at the imiss file. The last column shows the missing rate for samples. Since we used part of the 1000 Genome Project data this time, there are no missing samples or snps. So the missing rate is zero.</p>
<pre><code class="language-bash">head plink_results.imiss
 FID       IID MISS_PHENO   N_MISS   N_GENO   F_MISS
   0   HG00403          Y        0  1122299        0
   0   HG00404          Y        0  1122299        0
   0   HG00406          Y        0  1122299        0
   0   HG00407          Y        0  1122299        0
   0   HG00409          Y        0  1122299        0
   0   HG00410          Y        0  1122299        0
   0   HG00419          Y        0  1122299        0
   0   HG00421          Y        0  1122299        0
   0   HG00422          Y        0  1122299        0
</code></pre>
<p>For the meaning of headers, please refer to <a href="https://www.cog-genomics.org/plink/1.9/formats">PLINK documents</a>.</p>
<h3 id="allele-frequency">Allele Frequency</h3>
<p>One of the most important statistics of SNPs is their frequencies in a certain population. A number of downstream analysis is based on investigating differences in allele frequencies.</p>
<p>Usually, variants can be categorized into 3 groups based on their Minor Allel Frequency (MAF):</p>
<ol>
<li><strong>Common variants</strong> : MAF&gt;=0.05</li>
<li><strong>Low-frequency variants</strong> : 0.01&lt;=MAF&lt;0.05</li>
<li><strong>Rare variants</strong> : MAF&lt;0.01</li>
</ol>
<p>For different downstream analyses, we might use different set of variants. For example, for PCA, we might use only common variants. For gene-based tests, we might use only rare variants.</p>
<p>Using PLINK1.9 we can easily calculate the MAF of variants in the input data.</p>
<pre><code class="language-bash">plink \
    --bfile ${genotypeFile} \
    --freq \
    --out plink_results
</code></pre>
<p>Next we use plink2 to run the same options to check the difference between the results.</p>
<pre><code class="language-bash">plink2 \
        --bfile ${genotypeFile} \
        --freq \
        --out plink_results
</code></pre>
<pre><code class="language-bash"># results from plink1.9
head plink_results.frq
 CHR              SNP   A1   A2          MAF  NCHROBS
   1      1:13273:G:C    C    G       0.0625     1008
   1      1:14599:T:A    A    T      0.08929     1008
   1      1:14604:A:G    G    A      0.08929     1008
   1      1:14930:A:G    G    A       0.4137     1008
   1      1:69897:T:C    T    C        0.124     1008
   1      1:86331:A:G    G    A       0.0873     1008
   1      1:91581:G:A    A    G        0.498     1008
   1     1:122872:T:G    G    T       0.2589     1008
   1     1:135163:C:T    T    C      0.09226     1008
</code></pre>
<pre><code class="language-bash"># results from plink2
head plink_results.afreq
#CHROM  ID  REF ALT ALT_FREQS   OBS_CT
1   1:13273:G:C G   C   0.0625  1008
1   1:14599:T:A T   A   0.0892857   1008
1   1:14604:A:G A   G   0.0892857   1008
1   1:14930:A:G A   G   0.41369 1008
1   1:69897:T:C T   C   0.875992    1008
1   1:86331:A:G A   G   0.0873016   1008
1   1:91581:G:A G   A   0.498016    1008
1   1:122872:T:G    T   G   0.258929    1008
1   1:135163:C:T    C   T   0.0922619   1008
</code></pre>
<p>We need pay attention to the concepts here.</p>
<p>In PLINK1.9, the concept here is minor (A1) and major(A2) allele, while in PLINK2 it is reference(REF) and altervative(ALT) allele.</p>
<ul>
<li><strong>Major / Minor</strong>: Major allele and minor allele are defined as the allele with highest and lower(or second highest for multiallelic variants) allele in a given population, respectively. So major allele and minor allele for a SNP might be different in two independent populations. The range for MAF(minor allele frequencies) is [0,0.5].</li>
<li><strong>Ref / Alt</strong>: Reference(REF) and altervative(ALT) alleles are simply determined by the allele on a reference genome. If we use the same reference genome, the reference(REF) and altervative(ALT) allele will be the same acorss populations. Reference allele could be major or minor in different populations. The range for alternative allele frequency is [0,1], since it could be major or minor allele in a given population.</li>
</ul>
<h3 id="inbreeding-f-coefficient">Inbreeding F coefficient</h3>
<p>Next we can check the heterozygosity F of samples (https://www.cog-genomics.org/plink/1.9/basic_stats#ibc) : </p>
<p><code>-het</code> option will computes observed and expected autosomal homozygous genotype counts for each sample. Usually, we need to exclude individuals with high or low heterozygosity coefficient, which suggests that the sample might be contaminated. </p>
<pre><code class="language-bash">plink \
    --bfile ${genotypeFile} \
    --het \
    --out plink_results
</code></pre>
<p>Check the output:</p>
<pre><code class="language-bash">head plink_results.het
 FID       IID       O(HOM)       E(HOM)        N(NM)            F
   0   HG00403       747270    7.488e+05      1122299    -0.004114
   0   HG00404       748955    7.488e+05      1122299    0.0003974
   0   HG00406       750093    7.488e+05      1122299     0.003444
   0   HG00407       746566    7.488e+05      1122299    -0.005999
   0   HG00409       751694    7.488e+05      1122299     0.007731
   0   HG00410       745078    7.488e+05      1122299    -0.009983
   0   HG00419       747996    7.488e+05      1122299     -0.00217
   0   HG00421       757199    7.488e+05      1122299      0.02247
   0   HG00422       752725    7.488e+05      1122299      0.01049
</code></pre>
<p>A commonly used method is to exclude samples with heterzygosity F deviating more than 3 standard dividation(SD) from the mean F.</p>
<h3 id="hardy-weinberg-equilibrium-exact-test">Hardy-Weinberg equilibrium exact test</h3>
<p>For SNP QC, besides checking the missing rate, we also need to check if the SNP is in hardy-weinberg equilibrium:</p>
<p><code>--hardy</code> will perform Hardy-Weinberg equilibrium exact test for each variant. Variants with low P value usually suggest genotyping errors, or indicate evolutionary selection for these variants.</p>
<p>The following command can calculate the Hardy-Weinberg equilibrium exact test statistics for all SNPs. (https://www.cog-genomics.org/plink/1.9/basic_stats#hardy)</p>
<pre><code class="language-bash">plink \
    --bfile ${genotypeFile} \
    --hardy \
    --out plink_results
</code></pre>
<pre><code class="language-bash">head plink_results.hwe
 CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P 
   1      1:13273:G:C  ALL(NP)    C    G             1/61/442    0.121   0.1172       0.7113
   1      1:14599:T:A  ALL(NP)    A    T             1/88/415   0.1746   0.1626       0.1625
   1      1:14604:A:G  ALL(NP)    G    A             1/88/415   0.1746   0.1626       0.1625
   1      1:14930:A:G  ALL(NP)    G    A             4/409/91   0.8115   0.4851    1.679e-61
   1      1:69897:T:C  ALL(NP)    T    C            7/111/386   0.2202   0.2173            1
   1      1:86331:A:G  ALL(NP)    G    A             0/88/416   0.1746   0.1594      0.02387
   1      1:91581:G:A  ALL(NP)    A    G          137/228/139   0.4524      0.5      0.03271
   1     1:122872:T:G  ALL(NP)    G    T            1/259/244   0.5139   0.3838     8.04e-19
   1     1:135163:C:T  ALL(NP)    T    C             1/91/412   0.1806   0.1675       0.1066
</code></pre>
<h3 id="applying-filters">Applying filters</h3>
<p>Previously we just calculate the basic statistics using PLINK. But when performing certain analyses, we just want to exclude the bad quallity samples or SNPs instead of calculating the statistics for all samples and SNPs.</p>
<p>In this case we can apply the following filters for example:</p>
<ul>
<li><code>--maf 0.01</code> : exlcude snps with maf&lt;0.01</li>
<li><code>--geno 0.01</code> :filters out all variants with missing rates exceeding 0.0</li>
<li><code>--mind 0.02</code> :filters out all samples with missing rates exceeding 0.02</li>
<li><code>--hwe 5e-6</code> : filters out all variants which have Hardy-Weinberg equilibrium exact test p-value below the provided threshold. NOTE: With case/control data, cases and missing phenotypes are normally ignored. (see https://www.cog-genomics.org/plink/1.9/filter#hwe)</li>
</ul>
<h3 id="pruning">Pruning</h3>
<p>There are ofter strong Linkage disequilibrium(LD) among SNPs, for some analysis we don't need all SNPs and we need to remove the redundant SNPs to avoid bias in genetic estimations. For example, for relatedness estimation, we will use only LD-pruned SNP set. </p>
<p>We can use <code>--indep-pairwise 50 5 0.2</code> to filter out those are in strong LD and keep only the independent SNPS.
Please check https://www.cog-genomics.org/plink/1.9/ld#indep for the meaning of each parameter.
Combined with the filters we just introduced, we can run:</p>
<pre><code class="language-bash">plink \
    --bfile ${genotypeFile} \
    --maf 0.01 \
    --geno 0.01 \
    --mind 0.02 \
    --hwe 5e-6 \
    --indep-pairwise 50 5 0.2 \
    --out plink_results
</code></pre>
<p>This command generates two outputs :  <code>plink_results.prune.in</code> and <code>plink_results.prune.out</code>
<code>plink_results.prune.in</code> is the independent set of SNPs we will use in the following analysis.
Let's take a look at this file. Basically it just contains one SNP id per line.</p>
<pre><code class="language-bash">head plink_results.prune.in
1:13273:G:C
1:14599:T:A
1:69897:T:C
1:86331:A:G
1:91581:G:A
1:135163:C:T
1:233473:C:G
1:532929:T:C
1:559480:C:T
1:565433:C:T
</code></pre>
<h3 id="sample-snp-filtering-extractexcludekeepremove">Sample &amp; SNP filtering (extract/exclude/keep/remove)</h3>
<p>Some time we will use only a subset of samples or snps included the original dataset. 
In this case, we can use <code>--extract</code> or <code>--exclude</code> to select or exclude SNPs from analysis, <code>--keep</code> or <code>--remove</code> to select or exclude samples.</p>
<p>For  <code>--keep</code> or <code>--remove</code> , the input is the filename of a sample FID and IID file.
For <code>--extract</code> or <code>--exclude</code> , the input is the filename of a snplist file.</p>
<pre><code class="language-bash">head plink_results.prune.in
1:13273:G:C
1:14599:T:A
1:69897:T:C
1:86331:A:G
1:91581:G:A
1:135163:C:T
1:233473:C:G
1:532929:T:C
1:559480:C:T
1:565433:C:T
</code></pre>
<h3 id="ibd-pi_hat">IBD / PI_HAT</h3>
<p><code>--genome</code> will estimate IBS/IBD. Usually for this analysis, we need to prune our data first since the strong LD will cause bias in the results.
(This step is extremelly computationally intensive)</p>
<p>Combined with the <code>--extract</code>, we can run:</p>
<pre><code class="language-bash">plink \
        --bfile ${genotypeFile} \
    --extract plink_results.prune.in \
        --genome \
        --out plink_results
</code></pre>
<p>PI_HAT is the IBD estimation. Please check https://www.cog-genomics.org/plink/1.9/ibd for more details.</p>
<pre><code class="language-bash">head plink_results.genome
 FID1     IID1 FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO
   0  HG00403   0  HG00404 OT     0  0.9800  0.0086  0.0114  0.0157  -1  0.749375  0.5531  2.0089
   0  HG00403   0  HG00406 OT     0  0.9751  0.0231  0.0018  0.0133  -1  0.748336  0.6309  2.0225
   0  HG00403   0  HG00407 OT     0  0.9801  0.0153  0.0046  0.0122  -1  0.748293  0.5045  2.0007
   0  HG00403   0  HG00409 OT     0  0.9807  0.0193  0.0000  0.0097  -1  0.747215  0.6014  2.0173
   0  HG00403   0  HG00410 OT     0  0.9744  0.0256  0.0000  0.0128  -1  0.748058  0.8621  2.0745
   0  HG00403   0  HG00419 OT     0  0.9650  0.0350  0.0000  0.0175  -1  0.747146  0.8080  2.0595
   0  HG00403   0  HG00421 OT     0  0.9842  0.0158  0.0000  0.0079  -1  0.746522  0.4557  1.9926
   0  HG00403   0  HG00422 OT     0  0.9832  0.0111  0.0056  0.0112  -1  0.748156  0.4208  1.9867
   0  HG00403   0  HG00428 OT     0  0.9929  0.0000  0.0071  0.0071  -1  0.746851  0.3115  1.9674
</code></pre>
<h3 id="ld-calculation">LD calculation</h3>
<p>We can also use our data to calculate LD between a pair of SNPs.</p>
<p><img alt="image" src="https://user-images.githubusercontent.com/40289485/161413586-d4f6e21e-f0c7-4c54-a703-bb060ec6913d.png" /></p>
<p><code>--chr</code> option allows us to include snps on a specific chromosome.
To calculate LD r2 for SNPs on chr22 , we can run:</p>
<pre><code class="language-bash">plink \
    --bfile ${genotypeFile} \
        --chr 22 \
        --r2 \
        --out plink_results
</code></pre>
<pre><code class="language-bash">head plink_results.ld
 CHR_A         BP_A             SNP_A  CHR_B         BP_B             SNP_B           R2 
    22     16053659   22:16053659:A:C     22     16067500   22:16067500:T:C      0.64577 
    22     16053862   22:16053862:C:T     22     16054454   22:16054454:C:T     0.987505 
    22     16053862   22:16053862:C:T     22     16055122   22:16055122:G:T     0.601708 
    22     16053863   22:16053863:G:A     22     16055070   22:16055070:G:A     0.931043 
    22     16054454   22:16054454:C:T     22     16055122   22:16055122:G:T     0.594216 
    22     16058767   22:16058767:A:G     22     16067462   22:16067462:C:T     0.266916 
    22     16067462   22:16067462:C:T     22     16069771   22:16069771:G:A     0.382323 
    22     16069771   22:16069771:G:A     22     16123813   22:16123813:G:A     0.291935 
    22     16143946   22:16143946:A:G     22     16155259   22:16155259:A:G     0.559165 
</code></pre>
<h3 id="data-management-make-bedrecode">Data management (make-bed/recode)</h3>
<p>By far the input data we use is in binary form, but sometimes we may want the text version.
<img alt="image" src="https://user-images.githubusercontent.com/40289485/161413659-a489b508-63c7-4166-9f5c-25a1a125109a.png" /></p>
<p>To convert the format we can run:</p>
<pre><code class="language-bash">#extract the 1000 samples with the pruned SNPs, and make a bed file.
plink \
    --bfile ${genotypeFile} \
    --extract plink_results.prune.in \
    --make-bed \
    --out plink_1000_pruned

#convert the bed/bim/fam to ped/map
plink \
        --bfile plink_1000_pruned \
        --recode \
        --out plink_1000_pruned
</code></pre>
<h2 id="exercise">Exercise</h2>
<ul>
<li>Follow this tutorial and type in the commands:</li>
<li>Calculate basic statistics for the simulated data.</li>
<li>
<p>Learn the meaning of each QC step.</p>
</li>
<li>
<p>Visualize the results of QC (using Python or R)</p>
</li>
<li>Draw the distribution of MAF.(histogram)</li>
<li>Draw the distribution of het.(histogram)</li>
<li>Try to briefly explain what you observe</li>
</ul>
<h2 id="additional-resources">Additional resources</h2>
<ul>
<li>Marees, A. T., de Kluiver, H., Stringer, S., Vorspan, F., Curis, E., Marie‐Claire, C., &amp; Derks, E. M. (2018). A tutorial on conducting genome‐wide association studies: Quality control and statistical analysis. International journal of methods in psychiatric research, 27(2), e1608.</li>
</ul>
<h2 id="reference">Reference</h2>
<ul>
<li>Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A., Bender, D., ... &amp; Sham, P. C. (2007). PLINK: a tool set for whole-genome association and population-based linkage analyses. The American journal of human genetics, 81(3), 559-575.</li>
<li>Chang, C. C., Chow, C. C., Tellier, L. C., Vattikuti, S., Purcell, S. M., &amp; Lee, J. J. (2015). Second-generation PLINK: rising to the challenge of larger and richer datasets. Gigascience, 4(1), s13742-015.</li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>GWASTutorial is licensed under the MIT license</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>

        <div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
